diff -Nuar ncurses-6.1.orig/configure ncurses-6.1/configure
--- ncurses-6.1.orig/configure	2019-05-08 23:04:13.588926826 +0200
+++ ncurses-6.1/configure	2019-05-08 23:04:47.330924959 +0200
@@ -17509,6 +17509,7 @@
 /* System header to define __stub macros and hopefully few prototypes,
     which can conflict with char $ac_func (); below.  */
 #include <assert.h>
+#include <stdio.h>
 /* Override any gcc2 internal prototype to avoid an error.  */
 #ifdef __cplusplus
 extern "C"
@@ -17527,7 +17528,7 @@
 #if defined (__stub_$ac_func) || defined (__stub___$ac_func)
 choke me
 #else
-f = $ac_func; /* workaround for ICC 12.0.3 */ if (f == 0) return 1;
+f = $ac_func; printf("meh"); /* workaround for ICC 12.0.3 */ if (f == 0) return 1;
 #endif
 
   ;
