diff -Nuar Python-3.6.5.orig/Makefile.pre.in Python-3.6.5/Makefile.pre.in
--- Python-3.6.5.orig/Makefile.pre.in	2019-05-25 17:33:47.726574264 +0200
+++ Python-3.6.5/Makefile.pre.in	2019-05-25 17:34:59.334570303 +0200
@@ -1066,8 +1066,6 @@
 
 QUICKTESTOPTS=	$(TESTOPTS) -x test_subprocess test_io test_lib2to3 \
 		test_multibytecodec test_urllib2_localnet test_itertools \
-		test_multiprocessing_fork test_multiprocessing_spawn \
-		test_multiprocessing_forkserver \
 		test_mailbox test_socket test_poll \
 		test_select test_zipfile test_concurrent_futures
 quicktest:	@DEF_MAKE_RULE@ platform
