diff -Nuar gjs-1.54.3.orig/configure gjs-1.54.3/configure
--- gjs-1.54.3.orig/configure	2019-05-27 23:01:29.715498602 +0200
+++ gjs-1.54.3/configure	2019-05-27 23:03:19.401492535 +0200
@@ -22158,7 +22158,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-    #include <signal.h>
+    #include <linux/signal.h>
 int
 main ()
 {
diff -Nuar gjs-1.54.3.orig/configure.ac gjs-1.54.3/configure.ac
--- gjs-1.54.3.orig/configure.ac	2018-11-12 18:16:04.000000000 +0100
+++ gjs-1.54.3/configure.ac	2019-05-27 23:03:05.762493289 +0200
@@ -155,7 +155,7 @@
   have_sigev_thread_id=no
   AC_MSG_CHECKING([for Linux SIGEV_THREAD_ID])
   AC_COMPILE_IFELSE([
-    AC_LANG_PROGRAM([[#include <signal.h>]], [return SIGEV_THREAD_ID;])
+    AC_LANG_PROGRAM([[#include <linux/signal.h>]], [return SIGEV_THREAD_ID;])
   ], [
     have_sigev_thread_id=yes
     AC_MSG_RESULT([yes])
