diff -Nuar gjs-1.54.3.orig/gjs/context.cpp gjs-1.54.3/gjs/context.cpp
--- gjs-1.54.3.orig/gjs/context.cpp	2018-11-12 17:43:50.000000000 +0100
+++ gjs-1.54.3/gjs/context.cpp	2019-05-30 13:45:44.527684132 +0200
@@ -23,7 +23,7 @@
 
 #include <config.h>
 
-#include <signal.h>
+#include <linux/signal.h>
 #include <sys/types.h>
 #include <unistd.h>
 
diff -Nuar gjs-1.54.3.orig/gjs/profiler.cpp gjs-1.54.3/gjs/profiler.cpp
--- gjs-1.54.3.orig/gjs/profiler.cpp	2018-11-12 17:43:50.000000000 +0100
+++ gjs-1.54.3/gjs/profiler.cpp	2019-05-30 13:45:25.099685207 +0200
@@ -26,7 +26,7 @@
 #include <algorithm>
 #include <errno.h>
 #include <memory>
-#include <signal.h>
+#include <linux/signal.h>
 #include <sys/syscall.h>
 #include <sys/types.h>
 #include <unistd.h>
