diff -Nuar cfe-7.0.1.src.orig/include/clang/Driver/Distro.h cfe-7.0.1.src/include/clang/Driver/Distro.h
--- cfe-7.0.1.src.orig/include/clang/Driver/Distro.h	2018-05-10 10:45:43.000000000 +0200
+++ cfe-7.0.1.src/include/clang/Driver/Distro.h	2019-01-09 20:07:58.383013839 +0100
@@ -35,6 +35,7 @@
     DebianStretch,
     DebianBuster,
     Exherbo,
+    GentooLinux,
     RHEL5,
     RHEL6,
     RHEL7,
@@ -101,6 +102,10 @@
   /// @}
   /// @name Convenience Predicates
   /// @{
+  
+  bool IsGentooLinux() const {
+    return DistroVal == GentooLinux;
+  }
 
   bool IsRedhat() const {
     return DistroVal == Fedora || (DistroVal >= RHEL5 && DistroVal <= RHEL7);
diff -Nuar cfe-7.0.1.src.orig/lib/Driver/Distro.cpp cfe-7.0.1.src/lib/Driver/Distro.cpp
--- cfe-7.0.1.src.orig/lib/Driver/Distro.cpp	2018-07-20 10:19:20.000000000 +0200
+++ cfe-7.0.1.src/lib/Driver/Distro.cpp	2019-01-09 20:08:56.383010631 +0100
@@ -136,6 +136,9 @@
   if (VFS.exists("/etc/arch-release"))
     return Distro::ArchLinux;
 
+  if (VFS.exists("/etc/gentoo-release"))
+    return Distro::GentooLinux;
+
   return Distro::UnknownDistro;
 }
 
diff -Nuar cfe-7.0.1.src.orig/lib/Driver/ToolChains/Linux.cpp cfe-7.0.1.src/lib/Driver/ToolChains/Linux.cpp
--- cfe-7.0.1.src.orig/lib/Driver/ToolChains/Linux.cpp	2019-01-09 20:03:43.938027913 +0100
+++ cfe-7.0.1.src/lib/Driver/ToolChains/Linux.cpp	2019-01-09 20:06:30.160018719 +0100
@@ -233,7 +233,7 @@
     ExtraOpts.push_back("now");
   }
 
-  if (Distro.IsOpenSUSE() || Distro.IsUbuntu() || Distro.IsAlpineLinux()) {
+  if (Distro.IsOpenSUSE() || Distro.IsUbuntu() || Distro.IsAlpineLinux() || Distro.IsGentooLinux()) {
     ExtraOpts.push_back("-z");
     ExtraOpts.push_back("relro");
   }
