From 507de543f8ef73168685cd7479fd680e9f955b40 Mon Sep 17 00:00:00 2001
From: Steven Powell <spowellgrr@gmail.com>
Date: Wed, 2 Nov 2022 08:30:27 -0700
Subject: [PATCH] revert setting snapshotter to fuse-overlayfs

---
 deploy/kicbase/entrypoint | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/deploy/kicbase/entrypoint b/deploy/kicbase/entrypoint
index de3e797a5cda..b94ecbea0641 100755
--- a/deploy/kicbase/entrypoint
+++ b/deploy/kicbase/entrypoint
@@ -121,13 +121,13 @@ configure_containerd() {
       # if fuse is present, use fuse-overlayfs, else fallback to native
       # we do not use the ZFS snapshotter because of skew issues vs the host
       if [[ -e /dev/fuse ]]; then
-        snapshotter="fuse-overlayfs"
+        # snapshotter="fuse-overlayfs" skipping temporarily: https://github.com/kubernetes/minikube/issues/15191
       else
         snapshotter="native"
       fi
     # fuse likely implies fuse-overlayfs, we should switch to fuse-overlayfs (or native)
     elif [[ "$container_filesystem" == 'fuseblk' ]]; then
-      snapshotter="fuse-overlayfs"
+      # snapshotter="fuse-overlayfs" skipping temporarily: https://github.com/kubernetes/minikube/issues/15191
     fi
   fi
 
