diff -Nuar kio-extras-19.12.3.orig/smb/kio_smb.h kio-extras-19.12.3/smb/kio_smb.h
--- kio-extras-19.12.3.orig/smb/kio_smb.h	2020-03-03 20:33:29.000000000 +0100
+++ kio-extras-19.12.3/smb/kio_smb.h	2020-03-09 21:09:56.297094366 +0100
@@ -79,7 +79,7 @@
 //---------------------------
 #include "kio_smb_internal.h"
 
-#define MAX_XFER_BUF_SIZE           65534
+#define MAX_XFER_BUF_SIZE           4194303
 
 // Categorized logger
 Q_DECLARE_LOGGING_CATEGORY(KIO_SMB)
