diff -Nuar firefox-60.0.orig/build/unix/elfhack/moz.build firefox-60.0/build/unix/elfhack/moz.build
--- firefox-60.0.orig/build/unix/elfhack/moz.build	2018-05-03 18:58:26.000000000 +0200
+++ firefox-60.0/build/unix/elfhack/moz.build	2018-05-17 21:59:31.288492960 +0200
@@ -17,6 +17,9 @@
     if '-flto' in CONFIG['OS_CFLAGS']:
         SOURCES['test-array.c'].flags += ['-fno-lto']
         SOURCES['test-ctors.c'].flags += ['-fno-lto']
+        if '-fwhole-program-vtables' in CONFIG['OS_CFLAGS']:
+           SOURCES['test-array.c'].flags += ['-fno-whole-program-vtables']
+           SOURCES['test-ctors.c'].flags += ['-fno-whole-program-vtables']
 
 HOST_SOURCES += [
     'elf.cpp',
diff -Nuar firefox-60.0.orig/js/src/ctypes/libffi/configure firefox-60.0/js/src/ctypes/libffi/configure
--- firefox-60.0.orig/js/src/ctypes/libffi/configure	2018-05-17 21:56:56.205501539 +0200
+++ firefox-60.0/js/src/ctypes/libffi/configure	2018-05-17 22:00:43.923488943 +0200
@@ -14079,6 +14079,7 @@
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-60.0.orig/js/src/ctypes/libffi/m4/libtool.m4 firefox-60.0/js/src/ctypes/libffi/m4/libtool.m4
--- firefox-60.0.orig/js/src/ctypes/libffi/m4/libtool.m4	2018-05-03 18:58:27.000000000 +0200
+++ firefox-60.0/js/src/ctypes/libffi/m4/libtool.m4	2018-05-17 22:00:05.009491095 +0200
@@ -6992,6 +6992,7 @@
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-60.0.orig/modules/freetype2/builds/unix/aclocal.m4 firefox-60.0/modules/freetype2/builds/unix/aclocal.m4
--- firefox-60.0.orig/modules/freetype2/builds/unix/aclocal.m4	2018-05-03 18:58:31.000000000 +0200
+++ firefox-60.0/modules/freetype2/builds/unix/aclocal.m4	2018-05-17 22:03:43.489479010 +0200
@@ -7516,6 +7516,7 @@
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-60.0.orig/toolkit/library/moz.build firefox-60.0/toolkit/library/moz.build
--- firefox-60.0.orig/toolkit/library/moz.build	2018-05-17 21:56:53.296501699 +0200
+++ firefox-60.0/toolkit/library/moz.build	2018-05-17 22:03:14.024480640 +0200
@@ -92,10 +92,10 @@
 SOURCES['StaticXULComponentsStart.cpp'].no_pgo = True
 
 # Don't let LTO reorder StaticXULComponentsStart.o.
-for f in CONFIG['OS_CXXFLAGS']:
-    if f.startswith('-flto'):
-        SOURCES['StaticXULComponentsStart.cpp'].flags += ['-fno-lto']
-        break
+if '-flto' in CONFIG['OS_CXXFLAGS']:
+    SOURCES['StaticXULComponentsStart.cpp'].flags += ['-fno-lto']
+    if '-fwhole-program-vtables' in CONFIG['OS_CXXFLAGS']:
+        SOURCES['StaticXULComponentsStart.cpp'].flags += ['-fno-whole-program-vtables']
 
 if CONFIG['OS_ARCH'] == 'WINNT':
     SOURCES += [
diff -Nuar firefox-60.0.orig/toolkit/library/StaticXULComponentsEnd/moz.build firefox-60.0/toolkit/library/StaticXULComponentsEnd/moz.build
--- firefox-60.0.orig/toolkit/library/StaticXULComponentsEnd/moz.build	2018-05-03 18:58:41.000000000 +0200
+++ firefox-60.0/toolkit/library/StaticXULComponentsEnd/moz.build	2018-05-17 22:02:33.320482892 +0200
@@ -7,10 +7,10 @@
 ]
 
 # Don't let LTO reorder StaticXULComponentsStart.o.
-for f in CONFIG['OS_CXXFLAGS']:
-    if f.startswith('-flto'):
-        SOURCES['StaticXULComponentsEnd.cpp'].flags += ['-fno-lto']
-        break
+if '-flto' in CONFIG['OS_CXXFLAGS']:
+    SOURCES['StaticXULComponentsEnd.cpp'].flags += ['-fno-lto']
+    if '-fwhole-program-vtables' in CONFIG['OS_CXXFLAGS']:
+        SOURCES['StaticXULComponentsEnd.cpp'].flags += ['-fno-whole-program-vtables']
 
 Library('StaticXULComponentsEnd')
 
