diff -Nuar firefox-62.0.orig/build/unix/elfhack/moz.build firefox-62.0/build/unix/elfhack/moz.build
--- firefox-62.0.orig/build/unix/elfhack/moz.build	2018-08-30 18:44:17.000000000 +0200
+++ firefox-62.0/build/unix/elfhack/moz.build	2018-09-15 11:03:00.226222771 +0200
@@ -18,6 +18,9 @@
         if f.startswith('-flto'):
             SOURCES['test-array.c'].flags += ['-fno-lto']
             SOURCES['test-ctors.c'].flags += ['-fno-lto']
+        if f.startswith('-fwhole-program-vtables'):
+            SOURCES['test-array.c'].flags += ['-fno-whole-program-vtables']
+            SOURCES['test-ctors.c'].flags += ['-fno-whole-program-vtables']
 
 HOST_SOURCES += [
     'elf.cpp',
diff -Nuar firefox-62.0.orig/js/src/ctypes/libffi/configure firefox-62.0/js/src/ctypes/libffi/configure
--- firefox-62.0.orig/js/src/ctypes/libffi/configure	2018-09-15 11:01:09.113228917 +0200
+++ firefox-62.0/js/src/ctypes/libffi/configure	2018-09-15 11:02:11.982225439 +0200
@@ -14079,6 +14079,7 @@
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-62.0.orig/js/src/ctypes/libffi/m4/libtool.m4 firefox-62.0/js/src/ctypes/libffi/m4/libtool.m4
--- firefox-62.0.orig/js/src/ctypes/libffi/m4/libtool.m4	2018-08-30 18:44:20.000000000 +0200
+++ firefox-62.0/js/src/ctypes/libffi/m4/libtool.m4	2018-09-15 11:02:11.984225439 +0200
@@ -6992,6 +6992,7 @@
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-62.0.orig/modules/freetype2/builds/unix/aclocal.m4 firefox-62.0/modules/freetype2/builds/unix/aclocal.m4
--- firefox-62.0.orig/modules/freetype2/builds/unix/aclocal.m4	2018-08-30 18:44:27.000000000 +0200
+++ firefox-62.0/modules/freetype2/builds/unix/aclocal.m4	2018-09-15 11:02:11.987225439 +0200
@@ -7516,6 +7516,7 @@
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-62.0.orig/toolkit/library/moz.build firefox-62.0/toolkit/library/moz.build
--- firefox-62.0.orig/toolkit/library/moz.build	2018-09-15 11:01:06.407229067 +0200
+++ firefox-62.0/toolkit/library/moz.build	2018-09-15 11:05:29.722214502 +0200
@@ -100,7 +100,8 @@
 for f in CONFIG['OS_CXXFLAGS']:
     if f.startswith('-flto'):
         SOURCES['StaticXULComponentsStart.cpp'].flags += ['-fno-lto']
-        break
+    if f.startswith('-fwhole-program-vtables'):
+        SOURCES['StaticXULComponentsStart.cpp'].flags += ['-fno-whole-program-vtables']
 
 if CONFIG['OS_ARCH'] == 'WINNT':
     SOURCES += [
diff -Nuar firefox-62.0.orig/toolkit/library/StaticXULComponentsEnd/moz.build firefox-62.0/toolkit/library/StaticXULComponentsEnd/moz.build
--- firefox-62.0.orig/toolkit/library/StaticXULComponentsEnd/moz.build	2018-08-30 18:44:35.000000000 +0200
+++ firefox-62.0/toolkit/library/StaticXULComponentsEnd/moz.build	2018-09-15 11:06:46.794210239 +0200
@@ -10,7 +10,8 @@
 for f in CONFIG['OS_CXXFLAGS']:
     if f.startswith('-flto'):
         SOURCES['StaticXULComponentsEnd.cpp'].flags += ['-fno-lto']
-        break
+    if f.startswith('-fwhole-program-vtables'):
+        SOURCES['StaticXULComponentsEnd.cpp'].flags += ['-fno-whole-program-vtables']
 
 Library('StaticXULComponentsEnd')
 
