diff -Nuar firefox-55.0b7.orig/nsprpub/configure firefox-55.0b7/nsprpub/configure
--- firefox-55.0b7.orig/nsprpub/configure	2017-07-11 21:35:59.323989842 +0200
+++ firefox-55.0b7/nsprpub/configure	2017-07-11 22:07:30.770885220 +0200
@@ -5485,9 +5485,14 @@
 $as_echo "$result" >&6; }
 
 if test $result = "yes"; then
-   PROFILE_GEN_LDFLAGS="-fprofile-generate"
-   PROFILE_USE_CFLAGS="-fprofile-use -fprofile-correction -Wcoverage-mismatch"
-   PROFILE_USE_LDFLAGS="-fprofile-use"
+  if test -n "${CLANG_CC}"; then
+    PROFILE_GEN_CFLAGS="-fprofile-generate"
+    PROFILE_USE_CFLAGS="-fprofile-instr-use"
+  else
+    PROFILE_GEN_LDFLAGS="-fprofile-generate"
+    PROFILE_USE_CFLAGS="-fprofile-use -fprofile-correction -Wcoverage-mismatch"
+    PROFILE_USE_LDFLAGS="-fprofile-use"
+  fi
 fi
 
 CFLAGS="$_SAVE_CFLAGS"
diff -Nuar firefox-55.0b7.orig/old-configure firefox-55.0b7/old-configure
--- firefox-55.0b7.orig/old-configure	2017-07-06 22:09:43.000000000 +0200
+++ firefox-55.0b7/old-configure	2017-07-11 22:05:32.123891783 +0200
@@ -16842,9 +16842,14 @@
 echo "$ac_t""$result" 1>&6
 
 if test $result = "yes"; then
-  PROFILE_GEN_LDFLAGS="-fprofile-generate"
-  PROFILE_USE_CFLAGS="-fprofile-use -fprofile-correction -Wcoverage-mismatch"
-  PROFILE_USE_LDFLAGS="-fprofile-use"
+  if test -n "${CLANG_CC}"; then
+    PROFILE_GEN_CFLAGS="-fprofile-generate"
+    PROFILE_USE_CFLAGS="-fprofile-instr-use"
+  else
+    PROFILE_GEN_LDFLAGS="-fprofile-generate"
+    PROFILE_USE_CFLAGS="-fprofile-use -fprofile-correction -Wcoverage-mismatch"
+    PROFILE_USE_LDFLAGS="-fprofile-use"
+  fi
 fi
 
 CFLAGS="$_SAVE_CFLAGS"
