diff -Nuar firefox-55.0b7.orig/build/unix/elfhack/moz.build firefox-55.0b7/build/unix/elfhack/moz.build
--- firefox-55.0b7.orig/build/unix/elfhack/moz.build	2016-09-19 18:19:29.000000000 +0200
+++ firefox-55.0b7/build/unix/elfhack/moz.build	2017-07-10 22:16:16.576160646 +0200
@@ -17,6 +17,9 @@
     if '-flto' in CONFIG['OS_CFLAGS']:
         SOURCES['test-array.c'].flags += ['-fno-lto']
         SOURCES['test-ctors.c'].flags += ['-fno-lto']
+        if '-fwhole-program-vtables' in CONFIG['OS_CFLAGS']:
+            SOURCES['test-array.c'].flags += ['-fno-whole-program-vtables']
+            SOURCES['test-ctors.c'].flags += ['-fno-whole-program-vtables']
 
 HOST_SOURCES += [
     'elf.cpp',
diff -Nuar firefox-55.0b7.orig/js/src/ctypes/libffi/configure firefox-55.0b7/js/src/ctypes/libffi/configure
--- firefox-55.0b7.orig/js/src/ctypes/libffi/configure	2017-07-10 22:07:46.545188858 +0200
+++ firefox-55.0b7/js/src/ctypes/libffi/configure	2017-07-10 22:13:44.252169072 +0200
@@ -14078,7 +14078,8 @@
 
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
-*\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto -fno-whole-program-vtables" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-55.0b7.orig/js/src/ctypes/libffi/m4/libtool.m4 firefox-55.0b7/js/src/ctypes/libffi/m4/libtool.m4
--- firefox-55.0b7.orig/js/src/ctypes/libffi/m4/libtool.m4	2016-05-12 19:04:59.000000000 +0200
+++ firefox-55.0b7/js/src/ctypes/libffi/m4/libtool.m4	2017-07-10 22:14:12.664167500 +0200
@@ -6992,6 +6992,7 @@
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-55.0b7.orig/modules/freetype2/builds/unix/aclocal.m4 firefox-55.0b7/modules/freetype2/builds/unix/aclocal.m4
--- firefox-55.0b7.orig/modules/freetype2/builds/unix/aclocal.m4	2017-01-27 01:16:09.000000000 +0100
+++ firefox-55.0b7/modules/freetype2/builds/unix/aclocal.m4	2017-07-10 22:15:39.241162712 +0200
@@ -7516,6 +7516,7 @@
 _lt_libdeps_save_CFLAGS=$CFLAGS
 case "$CC $CFLAGS " in #(
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
+*\ -fwhole-program-vtables*\ *) CFLAGS="$CFLAGS -fno-whole-program-vtables" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
diff -Nuar firefox-55.0b7.orig/toolkit/library/moz.build firefox-55.0b7/toolkit/library/moz.build
--- firefox-55.0b7.orig/toolkit/library/moz.build	2017-07-10 22:07:42.666189072 +0200
+++ firefox-55.0b7/toolkit/library/moz.build	2017-07-10 22:14:43.160165814 +0200
@@ -83,6 +83,8 @@
 # Don't let LTO reorder StaticXULComponentsStart.o.
 if '-flto' in CONFIG['OS_CXXFLAGS']:
     SOURCES['StaticXULComponentsStart.cpp'].flags += ['-fno-lto']
+    if '-fwhole-program-vtables' in CONFIG['OS_CXXFLAGS']:
+        SOURCES['StaticXULComponentsStart.cpp'].flags += ['-fno-whole-program-vtables']
 
 if CONFIG['OS_ARCH'] == 'WINNT':
     SOURCES += [
diff -Nuar firefox-55.0b7.orig/toolkit/library/StaticXULComponentsEnd/moz.build firefox-55.0b7/toolkit/library/StaticXULComponentsEnd/moz.build
--- firefox-55.0b7.orig/toolkit/library/StaticXULComponentsEnd/moz.build	2016-05-12 19:05:09.000000000 +0200
+++ firefox-55.0b7/toolkit/library/StaticXULComponentsEnd/moz.build	2017-07-10 22:15:14.200164097 +0200
@@ -9,6 +9,8 @@
 # Don't let LTO reorder StaticXULComponentsStart.o.
 if '-flto' in CONFIG['OS_CXXFLAGS']:
     SOURCES['StaticXULComponentsEnd.cpp'].flags += ['-fno-lto']
+    if '-fwhole-program-vtables' in CONFIG['OS_CXXFLAGS']:
+        SOURCES['StaticXULComponentsEnd.cpp'].flags += ['-fno-whole-program-vtables']
 
 Library('StaticXULComponentsEnd')
 
