--- linux-5.11.orig/drivers/hwmon/amd_energy.c	2021-02-14 23:32:24.000000000 +0100
+++ linux-5.11/drivers/hwmon/amd_energy.c	2021-03-29 21:21:57.263819644 +0200
@@ -331,7 +331,7 @@
 static struct platform_device *amd_energy_platdev;
 
 static const struct x86_cpu_id cpu_ids[] __initconst = {
-	X86_MATCH_VENDOR_FAM_MODEL(AMD, 0x17, 0x31, NULL),
+	X86_MATCH_VENDOR_FAM_MODEL(AMD, 0x17, X86_MODEL_ANY, NULL),
 	X86_MATCH_VENDOR_FAM_MODEL(AMD, 0x19, 0x01, NULL),
 	{}
 };
