diff -Nuar linux-5.5.0-gentoo.orig/kernel/rcu/tree_plugin.h linux-5.5.0-gentoo/kernel/rcu/tree_plugin.h
--- linux-5.5.0-gentoo.orig/kernel/rcu/tree_plugin.h	2020-01-27 01:23:03.000000000 +0100
+++ linux-5.5.0-gentoo/kernel/rcu/tree_plugin.h	2020-01-27 21:39:01.364587796 +0100
@@ -360,7 +360,7 @@
 		WARN_ON_ONCE(current->rcu_read_lock_nesting > RCU_NEST_PMAX);
 	barrier();  /* critical section after entry code. */
 }
-EXPORT_SYMBOL_GPL(__rcu_read_lock);
+EXPORT_SYMBOL(__rcu_read_lock);
 
 /*
  * Preemptible RCU implementation for rcu_read_unlock().
@@ -390,7 +390,7 @@
 		WARN_ON_ONCE(rrln < 0 && rrln > RCU_NEST_NMAX);
 	}
 }
-EXPORT_SYMBOL_GPL(__rcu_read_unlock);
+EXPORT_SYMBOL(__rcu_read_unlock);
 
 /*
  * Advance a ->blkd_tasks-list pointer to the next entry, instead
