diff -Nuar glibc-2.24.orig/sysdeps/x86_64/mempcpy_chk.S glibc-2.24/sysdeps/x86_64/mempcpy_chk.S
--- glibc-2.24.orig/sysdeps/x86_64/mempcpy_chk.S	2016-08-02 04:01:36.000000000 +0200
+++ glibc-2.24/sysdeps/x86_64/mempcpy_chk.S	2017-08-06 11:00:11.726833080 +0200
@@ -19,7 +19,7 @@
 #include <sysdep.h>
 #include "asm-syntax.h"
 
-#ifndef PIC
+#ifndef SHARED
 	/* For libc.so this is defined in memcpy.S.
 	   For libc.a, this is a separate source to avoid
 	   mempcpy bringing in __chk_fail and all routines
