--- ROCT-Thunk-Interface-rocm-6.1.2.orig/tests/kfdtest/src/Assemble.cpp	2024-03-31 21:24:47.000000000 +0200
+++ ROCT-Thunk-Interface-rocm-6.1.2/tests/kfdtest/src/Assemble.cpp	2024-11-30 22:06:35.192144055 +0100
@@ -69,6 +69,9 @@
 #else
 #include <llvm/Support/TargetRegistry.h>
 #endif
+#if LLVM_VERSION_MAJOR > 18
+#include "llvm/Support/ManagedStatic.h"
+#endif
 
 #include <linux/elf.h>
 #include "OSWrapper.hpp"
