diff -Nuar zfs-0.8.4.orig/include/spl/sys/sysmacros.h zfs-0.8.4/include/spl/sys/sysmacros.h
--- zfs-0.8.4.orig/include/spl/sys/sysmacros.h	2020-05-05 18:56:29.718370651 +0200
+++ zfs-0.8.4/include/spl/sys/sysmacros.h	2020-05-14 13:12:14.434136140 +0200
@@ -27,6 +27,7 @@
 
 #include <linux/module.h>
 #include <linux/sched.h>
+#include <linux/sched/isolation.h>
 #include <linux/cpumask.h>
 #include <sys/debug.h>
 #include <sys/zone.h>
@@ -78,6 +79,7 @@
 #define	curproc				current
 #define	max_ncpus			num_possible_cpus()
 #define	boot_ncpus			num_online_cpus()
+#define	avail_cpus			housekeeping_cpumask(HK_FLAG_DOMAIN)
 #define	CPU_SEQID			smp_processor_id()
 #define	is_system_labeled()		0
 
diff -Nuar zfs-0.8.4.orig/module/spl/spl-taskq.c zfs-0.8.4/module/spl/spl-taskq.c
--- zfs-0.8.4.orig/module/spl/spl-taskq.c	2020-05-05 18:57:28.443960905 +0200
+++ zfs-0.8.4/module/spl/spl-taskq.c	2020-05-14 13:11:52.090135784 +0200
@@ -994,7 +994,7 @@
 	}
 
 	if (spl_taskq_thread_bind) {
-		last_used_cpu = (last_used_cpu + 1) % num_online_cpus();
+		last_used_cpu = cpumask_next_wrap(last_used_cpu, avail_cpus, last_used_cpu, true);
 		kthread_bind(tqt->tqt_thread, last_used_cpu);
 	}
 
