diff -Nuar lld-5.0.1.src.orig/ELF/Options.td lld-5.0.1.src/ELF/Options.td
--- lld-5.0.1.src.orig/ELF/Options.td	2017-08-10 22:04:41.000000000 +0200
+++ lld-5.0.1.src/ELF/Options.td	2018-02-10 11:23:24.988880634 +0100
@@ -347,6 +347,7 @@
 def alias_version_script_eq: J<"version-script=">, Alias<version_script>;
 def alias_version_V: Flag<["-"], "V">, Alias<version>;
 def alias_wrap_wrap: J<"wrap=">, Alias<wrap>;
+def alias_icf_safe: F<"icf=safe">, Alias<icf_all>;
 
 // Our symbol resolution algorithm handles symbols in archive files differently
 // than traditional linkers, so we don't need --start-group and --end-group.
