diff -Nuar lld-8.0.0.src.orig/ELF/Driver.cpp lld-8.0.0.src/ELF/Driver.cpp
--- lld-8.0.0.src.orig/ELF/Driver.cpp	2019-02-14 11:49:15.000000000 +0100
+++ lld-8.0.0.src/ELF/Driver.cpp	2019-03-27 22:19:22.900112949 +0100
@@ -801,7 +801,7 @@
   Config->LTODebugPassManager = Args.hasArg(OPT_lto_debug_pass_manager);
   Config->LTONewPassManager = Args.hasArg(OPT_lto_new_pass_manager);
   Config->LTONewPmPasses = Args.getLastArgValue(OPT_lto_newpm_passes);
-  Config->LTOO = args::getInteger(Args, OPT_lto_O, 2);
+  Config->LTOO = args::getInteger(Args, OPT_lto_O, 3);
   Config->LTOObjPath = Args.getLastArgValue(OPT_plugin_opt_obj_path_eq);
   Config->LTOPartitions = args::getInteger(Args, OPT_lto_partitions, 1);
   Config->LTOSampleProfile = Args.getLastArgValue(OPT_lto_sample_profile);
