diff -Nuar libutf8proc-2.2.0-1.orig/src/utf8proc_data.c libutf8proc-2.2.0-1/src/utf8proc_data.c
--- libutf8proc-2.2.0-1.orig/src/utf8proc_data.c	2018-08-23 19:19:20.000000000 +0200
+++ libutf8proc-2.2.0-1/src/utf8proc_data.c	2019-02-16 22:30:43.769946856 +0100
@@ -1,3 +1,5 @@
+#include "utf8proc.h"
+
 static const utf8proc_uint16_t utf8proc_sequences[] = {
   97, 98, 99, 100, 101, 102, 103, 
   104, 105, 106, 107, 108, 109, 110, 111, 
