diff -Nuar libutf8proc-2.2.0-1.orig/libutf8proc.pc.in libutf8proc-2.2.0-1/libutf8proc.pc.in
--- libutf8proc-2.2.0-1.orig/libutf8proc.pc.in	2018-08-23 19:19:20.000000000 +0200
+++ libutf8proc-2.2.0-1/libutf8proc.pc.in	2019-02-17 11:36:08.461757426 +0100
@@ -1,10 +1,10 @@
-prefix=PREFIX
+prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/LIBDIR
-includedir=${prefix}/INCLUDEDIR
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
 
 Name: libutf8proc
 Description: UTF8 processing
-Version: VERSION
+Version: @SO_MAJOR@.@SO_MINOR@.@SO_PATCH@
 Libs: -L${libdir} -lutf8proc
 Cflags: -I${includedir} -DUTF8PROC_EXPORTS
