diff -Nuar gnu-efi-3.0.17.orig/lib/ctors.S gnu-efi-3.0.17/lib/ctors.S
--- gnu-efi-3.0.17.orig/lib/ctors.S	2023-05-02 13:51:45.000000000 +0200
+++ gnu-efi-3.0.17/lib/ctors.S	2024-01-02 12:48:01.945025201 +0100
@@ -8,7 +8,7 @@
  * end/END definitions, and the fact that they're mergeable, they can also
  * have NULLs which aren't guaranteed to be at the end.
  */
-	.section .init_array, "aM", @init_array
+	.section .init_array, "aw", @init_array
 	.p2align 3, 0
 	.globl _init_array
 _init_array:
@@ -16,7 +16,7 @@
 	.globl _init_array_end
 _init_array_end:
 	.long 0
-	.section .ctors, "aM", @init_array
+	.section .ctors, "aw", @init_array
 	.p2align 3, 0
 	.globl __CTOR_LIST__
 __CTOR_LIST__:
@@ -24,7 +24,7 @@
 	.globl __CTOR_END__
 __CTOR_END__:
 	.long 0
-	.section .dtors, "aM", @fini_array
+	.section .dtors, "aw", @fini_array
 	.p2align 3, 0
 	.globl __DTOR_LIST__
 __DTOR_LIST__:
@@ -32,7 +32,7 @@
 	.globl __DTOR_END__
 __DTOR_END__:
 	.long 0
-	.section .fini_array, "aM", @fini_array
+	.section .fini_array, "aw", @fini_array
 	.p2align 3, 0
 	.globl _fini_array
 _fini_array:
