diff -Nuar graphviz-2.40.1.orig/configure.ac graphviz-2.40.1/configure.ac
--- graphviz-2.40.1.orig/configure.ac	2019-03-10 12:44:51.261474234 +0100
+++ graphviz-2.40.1/configure.ac	2019-03-10 12:47:13.327466376 +0100
@@ -391,7 +391,7 @@
 					;;
 				esac
 		    fi
-	        CFLAGS="${CFLAGS} -Wno-unknown-pragmas -Wstrict-prototypes -Wpointer-arith -Wall -ffast-math"
+	        CFLAGS="${CFLAGS} -Wno-unknown-pragmas -Wstrict-prototypes -Wpointer-arith -Wall"
 	        ;;
 	    esac
 	    ;;
diff -Nuar graphviz-2.40.1.orig/graphviz.spec.in graphviz-2.40.1/graphviz.spec.in
--- graphviz-2.40.1.orig/graphviz.spec.in	2016-12-05 00:18:35.000000000 +0100
+++ graphviz-2.40.1/graphviz.spec.in	2019-03-10 12:48:23.567462491 +0100
@@ -884,9 +884,6 @@
 #done
 
 %build
-# XXX ix86 only used to have -ffast-math, let's use everywhere
-%{expand: %%define optflags %{optflags} -ffast-math}
-
 # %%configure is broken in RH7.3 rpmbuild
 CFLAGS="$RPM_OPT_FLAGS" \
 ./configure \
