diff -Nuar chromium-66.0.3359.139.orig/build/config/compiler/BUILD.gn chromium-66.0.3359.139/build/config/compiler/BUILD.gn
--- chromium-66.0.3359.139.orig/build/config/compiler/BUILD.gn	2018-05-01 12:24:41.521827117 +0200
+++ chromium-66.0.3359.139/build/config/compiler/BUILD.gn	2018-05-01 12:27:15.656818591 +0200
@@ -561,10 +561,8 @@
       }
     }
 
-    # Disable optimization for now because they increase binary size by too
-    # much.
     if (use_lld && (is_android || (is_linux && !is_chromeos))) {
-      ldflags += [ "-Wl,--lto-O0" ]
+      ldflags += [ "-Wl,--lto-O3" ]
     }
 
     cflags += [ "-fwhole-program-vtables" ]
