--- chromium-73.0.3683.86.orig/build/config/compiler/BUILD.gn	2019-04-19 20:40:35.782062283 +0200
+++ chromium-73.0.3683.86/build/config/compiler/BUILD.gn	2019-04-19 20:45:51.178044837 +0200
@@ -135,8 +135,7 @@
   # TODO(gbiv): We disable optimizations by default on most platforms because
   # the space overhead is too great. We should use some mixture of profiles and
   # optimization settings to better tune the size increase.
-  thin_lto_enable_optimizations =
-      (is_chromeos || is_android) && is_official_build
+  thin_lto_enable_optimizations = true
 
   # By default only the binaries in official builds get build IDs.
   force_local_build_id = false
