diff -Nuar chromium-122.0.6261.57.orig/build/config/compiler/BUILD.gn chromium-122.0.6261.57/build/config/compiler/BUILD.gn
--- chromium-122.0.6261.57.orig/build/config/compiler/BUILD.gn	2024-02-22 23:07:20.344358587 +0100
+++ chromium-122.0.6261.57/build/config/compiler/BUILD.gn	2024-02-22 23:11:04.830397099 +0100
@@ -1042,7 +1042,7 @@
 config("thinlto_optimize_max") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
     if (thin_lto_enable_optimizations) {
-      lto_opt_level = 2
+      lto_opt_level = 3
     } else {
       lto_opt_level = 0
     }
