diff -Nuar chromium-71.0.3578.30.orig/build/config/compiler/BUILD.gn chromium-71.0.3578.30/build/config/compiler/BUILD.gn
--- chromium-71.0.3578.30.orig/build/config/compiler/BUILD.gn	2018-11-11 11:56:41.787966867 +0100
+++ chromium-71.0.3578.30/build/config/compiler/BUILD.gn	2018-11-11 12:01:23.207951301 +0100
@@ -642,7 +642,7 @@
         # Limit the size of the ThinLTO cache to the lesser of 10% of available disk
         # space, 10GB and 100000 files.
         cache_policy =
-            "cache_size=10%:cache_size_bytes=10g:cache_size_files=100000"
+            "cache_size=25%:cache_size_bytes=30g:cache_size_files=1000000"
         ldflags += [
           "-Wl,--thinlto-jobs=8",
           "-Wl,--thinlto-cache-dir=" +
