diff -Nuar webkitgtk-2.4.11.orig/Source/JavaScriptCore/API/JSStringRef.cpp webkitgtk-2.4.11/Source/JavaScriptCore/API/JSStringRef.cpp
--- webkitgtk-2.4.11.orig/Source/JavaScriptCore/API/JSStringRef.cpp	2016-04-10 08:48:36.000000000 +0200
+++ webkitgtk-2.4.11/Source/JavaScriptCore/API/JSStringRef.cpp	2017-09-20 10:49:36.472662767 +0200
@@ -37,7 +37,7 @@
 JSStringRef JSStringCreateWithCharacters(const JSChar* chars, size_t numChars)
 {
     initializeThreading();
-    return OpaqueJSString::create(chars, numChars).leakRef();
+    return OpaqueJSString::create(reinterpret_cast<const UChar*>(chars), numChars).leakRef();
 }
 
 JSStringRef JSStringCreateWithUTF8CString(const char* string)
@@ -62,7 +62,7 @@
 JSStringRef JSStringCreateWithCharactersNoCopy(const JSChar* chars, size_t numChars)
 {
     initializeThreading();
-    return OpaqueJSString::create(StringImpl::createWithoutCopying(chars, numChars)).leakRef();
+    return OpaqueJSString::create(StringImpl::createWithoutCopying(reinterpret_cast<const UChar*>(chars), numChars)).leakRef();
 }
 
 JSStringRef JSStringRetain(JSStringRef string)
@@ -83,7 +83,7 @@
 
 const JSChar* JSStringGetCharactersPtr(JSStringRef string)
 {
-    return string->characters();
+    return reinterpret_cast<const JSChar*>(string->characters());
 }
 
 size_t JSStringGetMaximumUTF8CStringSize(JSStringRef string)
diff -Nuar webkitgtk-2.4.11.orig/Source/JavaScriptCore/runtime/DateConversion.cpp webkitgtk-2.4.11/Source/JavaScriptCore/runtime/DateConversion.cpp
--- webkitgtk-2.4.11.orig/Source/JavaScriptCore/runtime/DateConversion.cpp	2013-08-03 18:10:38.000000000 +0200
+++ webkitgtk-2.4.11/Source/JavaScriptCore/runtime/DateConversion.cpp	2017-09-20 10:49:36.472662767 +0200
@@ -107,7 +107,8 @@
 #if OS(WINDOWS)
             TIME_ZONE_INFORMATION timeZoneInformation;
             GetTimeZoneInformation(&timeZoneInformation);
-            const WCHAR* timeZoneName = t.isDST() ? timeZoneInformation.DaylightName : timeZoneInformation.StandardName;
+            const WCHAR* winTimeZoneName = t.isDST() ? timeZoneInformation.DaylightName : timeZoneInformation.StandardName;
+            String timeZoneName(reinterpret_cast<const UChar*>(winTimeZoneName));
 #else
             struct tm gtm = t;
             char timeZoneName[70];
diff -Nuar webkitgtk-2.4.11.orig/Source/WebKit2/Shared/API/c/WKString.cpp webkitgtk-2.4.11/Source/WebKit2/Shared/API/c/WKString.cpp
--- webkitgtk-2.4.11.orig/Source/WebKit2/Shared/API/c/WKString.cpp	2016-04-10 08:48:37.000000000 +0200
+++ webkitgtk-2.4.11/Source/WebKit2/Shared/API/c/WKString.cpp	2017-09-20 10:49:36.473662767 +0200
@@ -55,7 +55,7 @@
 size_t WKStringGetCharacters(WKStringRef stringRef, WKChar* buffer, size_t bufferLength)
 {
     COMPILE_ASSERT(sizeof(WKChar) == sizeof(UChar), WKStringGetCharacters_sizeof_WKChar_matches_UChar);
-    return (toImpl(stringRef)->getCharacters(static_cast<UChar*>(buffer), bufferLength));
+    return (toImpl(stringRef)->getCharacters(reinterpret_cast<UChar*>(buffer), bufferLength));
 }
 
 size_t WKStringGetMaximumUTF8CStringSize(WKStringRef stringRef)
