diff -Nuar tbb-2018_U3.orig/build/linux.gcc.inc tbb-2018_U3/build/linux.gcc.inc
--- tbb-2018_U3.orig/build/linux.gcc.inc	2019-02-10 19:57:47.867065320 +0100
+++ tbb-2018_U3/build/linux.gcc.inc	2019-02-10 19:58:54.385061641 +0100
@@ -60,13 +60,6 @@
     INCLUDE_TEST_HEADERS = -include $(tbb_root)/src/test/harness_preload.h
 endif
 
-# gcc 6.0 and later have -flifetime-dse option that controls
-# elimination of stores done outside the object lifetime
-ifneq (,$(shell gcc -dumpversion | egrep  "^([6-9])"))
-    # keep pre-contruction stores for zero initialization
-    DSE_KEY = -flifetime-dse=1
-endif
-
 ifeq ($(cfg), release)
         CPLUS_FLAGS = $(ITT_NOTIFY) -g -DUSE_PTHREAD
 endif
