diff -Nuar bedup-9999.orig/bedup/platform/chattr.py bedup-9999/bedup/platform/chattr.py
--- bedup-9999.orig/bedup/platform/chattr.py	2016-04-09 11:06:49.207309666 +0200
+++ bedup-9999/bedup/platform/chattr.py	2016-04-09 11:55:33.356898678 +0200
@@ -33,10 +33,10 @@
 #define FS_IOC_GETFLAGS ...
 #define FS_IOC_SETFLAGS ...
 
-#define	FS_SECRM_FL ... /* Secure deletion */
-#define	FS_UNRM_FL ... /* Undelete */
-#define	FS_COMPR_FL ... /* Compress file */
-#define FS_SYNC_FL ... /* Synchronous updates */
+#define FS_SECRM_FL ... /* Secure deletion */
+#define FS_UNRM_FL ... /* Undelete */
+#define FS_COMPR_FL ... /* Compress file */
+#define FS_SYNC_FL ... /* Synchronous updates */
 #define FS_IMMUTABLE_FL ... /* Immutable file */
 #define FS_APPEND_FL ... /* writes to file may only append */
 #define FS_NODUMP_FL ... /* do not dump file */
@@ -45,8 +45,8 @@
 #define FS_DIRTY_FL ...
 #define FS_COMPRBLK_FL ... /* One or more compressed clusters */
 #define FS_NOCOMP_FL ... /* Don't compress */
-#define FS_ECOMPR_FL ... /* Compression error */
 /* End compression flags --- maybe not all used */
+#define FS_ENCRYPT_FL ... /* Encrypted file */
 #define FS_BTREE_FL ... /* btree format dir */
 #define FS_INDEX_FL ... /* hash-indexed directory */
 #define FS_IMAGIC_FL ... /* AFS directory */
@@ -54,9 +54,13 @@
 #define FS_NOTAIL_FL ... /* file tail should not be merged */
 #define FS_DIRSYNC_FL ... /* dirsync behaviour (directories only) */
 #define FS_TOPDIR_FL ... /* Top of directory hierarchies*/
+#define FS_HUGE_FILE_FL ... /* Reserved for ext4 */
 #define FS_EXTENT_FL ... /* Extents */
-#define FS_DIRECTIO_FL ... /* Use direct i/o */
+#define FS_EA_INODE_FL ... /* Inode used for large EA */
+#define FS_EOFBLOCKS_FL ... /* Reserved for ext4 */
 #define FS_NOCOW_FL ... /* Do not cow file */
+#define FS_INLINE_DATA_FL ... /* Reserved for ext4 */
+#define FS_PROJINHERIT_FL ... /* Create with parents projid */
 #define FS_RESERVED_FL ... /* reserved for ext2 lib */
 
 #define FS_FL_USER_VISIBLE ... /* User visible flags */
