diff -Nuar libunwind-1.2.1.orig/include/compiler.h libunwind-1.2.1/include/compiler.h
--- libunwind-1.2.1.orig/include/compiler.h	2016-11-18 04:28:29.000000000 +0100
+++ libunwind-1.2.1/include/compiler.h	2017-09-12 16:20:27.425880555 +0200
@@ -40,7 +40,7 @@
 # if (__GNUC__ > 3) || (__GNUC__ == 3 && __GNUC_MINOR__ > 2)
 #  define ALWAYS_INLINE inline __attribute__((always_inline))
 #  define HIDDEN        __attribute__((visibility ("hidden")))
-#  define PROTECTED     __attribute__((visibility ("protected")))
+#  define PROTECTED
 # else
 #  define ALWAYS_INLINE
 #  define HIDDEN
