diff -Nuar pulseaudio-12.2.orig/aclocal.m4 pulseaudio-12.2/aclocal.m4
--- pulseaudio-12.2.orig/aclocal.m4	2019-04-15 23:05:32.962427621 +0200
+++ pulseaudio-12.2/aclocal.m4	2019-04-15 23:06:26.379424667 +0200
@@ -847,7 +847,6 @@
 dnl Autoconf wants to disallow AM_ names.  We explicitly allow
 dnl the ones we care about.
 m4_pattern_allow([^AM_[A-Z]+FLAGS$])dnl
-AC_REQUIRE([AM_SET_CURRENT_AUTOMAKE_VERSION])dnl
 AC_REQUIRE([AC_PROG_INSTALL])dnl
 if test "`cd $srcdir && pwd`" != "`pwd`"; then
   # Use -I$(srcdir) only when $(srcdir) != ., so that make's output
