--- libcupsfilters-2.1.0.orig/Makefile.in	2024-10-17 22:58:19.000000000 +0200
+++ libcupsfilters-2.1.0/Makefile.in	2024-11-06 18:23:12.021749667 +0100
@@ -1153,8 +1153,7 @@
 	-lm -ldl -lcups
 
 testfilters_LDFLAGS = \
-	-D_GNU_SOURCE \
-	-L/usr/lib
+	-D_GNU_SOURCE
 
 
 # =========
--- libcupsfilters-2.1.0.orig/Makefile.am	2024-10-17 22:58:08.000000000 +0200
+++ libcupsfilters-2.1.0/Makefile.am	2024-11-06 18:22:50.361716085 +0100
@@ -326,8 +326,7 @@
 	-lm -ldl -lcups
 
 testfilters_LDFLAGS = \
-	-D_GNU_SOURCE \
-	-L/usr/lib
+	-D_GNU_SOURCE
 
 EXTRA_DIST += \
 	$(check_SCRIPTS) \
