diff -Nuar ddcutil-0.9.1.orig/src/vcp/vcp_feature_set.c ddcutil-0.9.1/src/vcp/vcp_feature_set.c
--- ddcutil-0.9.1.orig/src/vcp/vcp_feature_set.c	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/vcp/vcp_feature_set.c	2019-02-18 23:01:06.090866451 +0100
@@ -321,16 +321,6 @@
 #endif
 
 
-static inline struct vcp_feature_set *
-unopaque_feature_set(
-      VCP_Feature_Set feature_set)
-{
-   struct vcp_feature_set * fset = (struct vcp_feature_set *) feature_set;
-   assert( fset && memcmp(fset->marker, VCP_FEATURE_SET_MARKER, 4) == 0);
-   return fset;
-}
-
-
 void free_feature_set(VCP_Feature_Set feature_set) {
    struct vcp_feature_set * fset = (struct vcp_feature_set *) feature_set;
    assert( fset && memcmp(fset->marker, VCP_FEATURE_SET_MARKER, 4) == 0);
