diff -Nuar ddcutil-0.9.1.orig/src/adl/Makefile.am ddcutil-0.9.1/src/adl/Makefile.am
--- ddcutil-0.9.1.orig/src/adl/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/adl/Makefile.am	2019-02-18 23:16:32.033815234 +0100
@@ -6,7 +6,6 @@
 AM_CPPFLAGS +=  -I@ADL_HEADER_DIR@
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -Nuar ddcutil-0.9.1.orig/src/app_common/Makefile.am ddcutil-0.9.1/src/app_common/Makefile.am
--- ddcutil-0.9.1.orig/src/app_common/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/app_common/Makefile.am	2019-02-18 23:16:22.033815788 +0100
@@ -5,7 +5,6 @@
 
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -Nuar ddcutil-0.9.1.orig/src/app_sysenv/Makefile.am ddcutil-0.9.1/src/app_sysenv/Makefile.am
--- ddcutil-0.9.1.orig/src/app_sysenv/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/app_sysenv/Makefile.am	2019-02-18 23:16:13.945816235 +0100
@@ -15,7 +15,6 @@
 
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
diff -Nuar ddcutil-0.9.1.orig/src/base/Makefile.am ddcutil-0.9.1/src/base/Makefile.am
--- ddcutil-0.9.1.orig/src/base/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/base/Makefile.am	2019-02-18 23:16:03.889816791 +0100
@@ -4,7 +4,6 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
diff -Nuar ddcutil-0.9.1.orig/src/cmdline/Makefile.am ddcutil-0.9.1/src/cmdline/Makefile.am
--- ddcutil-0.9.1.orig/src/cmdline/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/cmdline/Makefile.am	2019-02-18 23:15:54.801817294 +0100
@@ -4,7 +4,6 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS  = -Wall 
-AM_CFLAGS += -Werror
 # flags g_option parser structs in cmd_parser_goption.c
 # AM_CFLAGS += -Wpedantic
 
diff -Nuar ddcutil-0.9.1.orig/src/ddc/Makefile.am ddcutil-0.9.1/src/ddc/Makefile.am
--- ddcutil-0.9.1.orig/src/ddc/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/ddc/Makefile.am	2019-02-18 23:15:45.313817819 +0100
@@ -11,7 +11,6 @@
 
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # Causes PROGRAM_LOGIC_ERROR() macro to fail:
 # AM_CFLAGS += -Wpedantic
 
diff -Nuar ddcutil-0.9.1.orig/src/i2c/Makefile.am ddcutil-0.9.1/src/i2c/Makefile.am
--- ddcutil-0.9.1.orig/src/i2c/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/i2c/Makefile.am	2019-02-18 23:15:36.833818288 +0100
@@ -4,7 +4,6 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
diff -Nuar ddcutil-0.9.1.orig/src/sample_clients/Makefile.am ddcutil-0.9.1/src/sample_clients/Makefile.am
--- ddcutil-0.9.1.orig/src/sample_clients/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/sample_clients/Makefile.am	2019-02-18 23:15:25.081818938 +0100
@@ -7,7 +7,6 @@
 -I$(top_srcdir)/src
 
 AM_CFLAGS = -Wall -fPIC
-AM_CFLAGS += -Werror
 
 check_PROGRAMS =
 if ENABLE_SHARED_LIB_COND
diff -Nuar ddcutil-0.9.1.orig/src/util/Makefile.am ddcutil-0.9.1/src/util/Makefile.am
--- ddcutil-0.9.1.orig/src/util/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/util/Makefile.am	2019-02-18 23:15:09.058819824 +0100
@@ -3,7 +3,6 @@
   $(LIBDRM_CFLAGS)
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # -pedantic issues warnings re code that doesn't conform to ISO C
 #  Alas, -m modifier on sscanf is a POSIX extension, not ISO C
 #  In combination with -Werror, causes device_util.c to fail compilation
diff -Nuar ddcutil-0.9.1.orig/src/vcp/Makefile.am ddcutil-0.9.1/src/vcp/Makefile.am
--- ddcutil-0.9.1.orig/src/vcp/Makefile.am	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/vcp/Makefile.am	2019-02-18 23:14:57.626820456 +0100
@@ -4,7 +4,6 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS  = -Wall 
-AM_CFLAGS += -Werror
 # vcp_feature_codes.c requires extensive changes if -Wpedantic
 # AM_CFLAGS += -Wpedantic
 
