diff -Nuar ddcutil-0.9.9.orig/src/adl/Makefile.am ddcutil-0.9.9/src/adl/Makefile.am
--- ddcutil-0.9.9.orig/src/adl/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/adl/Makefile.am	2020-10-20 20:49:57.818489795 +0200
@@ -6,7 +6,6 @@
 AM_CPPFLAGS +=  -I@ADL_HEADER_DIR@
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -Nuar ddcutil-0.9.9.orig/src/app_sysenv/Makefile.am ddcutil-0.9.9/src/app_sysenv/Makefile.am
--- ddcutil-0.9.9.orig/src/app_sysenv/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/app_sysenv/Makefile.am	2020-10-20 20:50:06.802462556 +0200
@@ -11,7 +11,6 @@
 
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
diff -Nuar ddcutil-0.9.9.orig/src/base/Makefile.am ddcutil-0.9.9/src/base/Makefile.am
--- ddcutil-0.9.9.orig/src/base/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/base/Makefile.am	2020-10-20 20:50:15.786435316 +0200
@@ -4,7 +4,6 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
diff -Nuar ddcutil-0.9.9.orig/src/cmdline/Makefile.am ddcutil-0.9.9/src/cmdline/Makefile.am
--- ddcutil-0.9.9.orig/src/cmdline/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/cmdline/Makefile.am	2020-10-20 20:50:22.730414262 +0200
@@ -4,7 +4,6 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS  = -Wall 
-AM_CFLAGS += -Werror
 # flags g_option parser structs in cmd_parser_goption.c
 # AM_CFLAGS += -Wpedantic
 
diff -Nuar ddcutil-0.9.9.orig/src/ddc/Makefile.am ddcutil-0.9.9/src/ddc/Makefile.am
--- ddcutil-0.9.9.orig/src/ddc/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/ddc/Makefile.am	2020-10-20 20:50:46.986340718 +0200
@@ -9,7 +9,6 @@
 endif
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # Causes PROGRAM_LOGIC_ERROR() macro to fail:
 # AM_CFLAGS += -Wpedantic
 
diff -Nuar ddcutil-0.9.9.orig/src/dynvcp/Makefile.am ddcutil-0.9.9/src/dynvcp/Makefile.am
--- ddcutil-0.9.9.orig/src/dynvcp/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/dynvcp/Makefile.am	2020-10-20 20:50:39.786362549 +0200
@@ -9,7 +9,6 @@
 endif
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # Causes PROGRAM_LOGIC_ERROR() macro to fail:
 # AM_CFLAGS += -Wpedantic
 
diff -Nuar ddcutil-0.9.9.orig/src/i2c/Makefile.am ddcutil-0.9.9/src/i2c/Makefile.am
--- ddcutil-0.9.9.orig/src/i2c/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/i2c/Makefile.am	2020-10-20 20:50:30.298391316 +0200
@@ -4,7 +4,6 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
diff -Nuar ddcutil-0.9.9.orig/src/sample_clients/Makefile.am ddcutil-0.9.9/src/sample_clients/Makefile.am
--- ddcutil-0.9.9.orig/src/sample_clients/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/sample_clients/Makefile.am	2020-10-20 20:51:22.610232706 +0200
@@ -7,7 +7,6 @@
 -I$(top_srcdir)/src
 
 AM_CFLAGS = -Wall -fPIC
-AM_CFLAGS += -Werror
 
 check_PROGRAMS =
 if ENABLE_SHARED_LIB_COND
diff -Nuar ddcutil-0.9.9.orig/src/test/Makefile.am ddcutil-0.9.9/src/test/Makefile.am
--- ddcutil-0.9.9.orig/src/test/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/test/Makefile.am	2020-10-20 20:51:31.273206440 +0200
@@ -4,7 +4,7 @@
 -I$(top_srcdir)/src \
 -I$(top_srcdir)/src/public 
 
-AM_CFLAGS = -Wall -Werror
+AM_CFLAGS = -Wall
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -Nuar ddcutil-0.9.9.orig/src/util/Makefile.am ddcutil-0.9.9/src/util/Makefile.am
--- ddcutil-0.9.9.orig/src/util/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/util/Makefile.am	2020-10-20 20:51:15.930252960 +0200
@@ -3,7 +3,6 @@
   $(LIBDRM_CFLAGS)
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
 # -pedantic issues warnings re code that doesn't conform to ISO C
 #  Alas, -m modifier on sscanf is a POSIX extension, not ISO C
 #  In combination with -Werror, causes device_util.c to fail compilation
diff -Nuar ddcutil-0.9.9.orig/src/vcp/Makefile.am ddcutil-0.9.9/src/vcp/Makefile.am
--- ddcutil-0.9.9.orig/src/vcp/Makefile.am	2020-06-20 18:05:47.000000000 +0200
+++ ddcutil-0.9.9/src/vcp/Makefile.am	2020-10-20 20:50:55.482314959 +0200
@@ -4,7 +4,6 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS  = -Wall 
-AM_CFLAGS += -Werror
 # vcp_feature_codes.c requires extensive changes if -Wpedantic
 # AM_CFLAGS += -Wpedantic
 
