diff -Nuar ddcutil-0.9.1.orig/src/test/i2c/i2c_io_old.c ddcutil-0.9.1/src/test/i2c/i2c_io_old.c
--- ddcutil-0.9.1.orig/src/test/i2c/i2c_io_old.c	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/test/i2c/i2c_io_old.c	2019-01-05 21:50:48.398750282 +0100
@@ -34,7 +34,6 @@
 
 #include <assert.h>
 #include <errno.h>
-#include <execinfo.h>
 #include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Nuar ddcutil-0.9.1.orig/src/util/debug_util.c ddcutil-0.9.1/src/util/debug_util.c
--- ddcutil-0.9.1.orig/src/util/debug_util.c	2018-05-25 16:22:17.000000000 +0200
+++ ddcutil-0.9.1/src/util/debug_util.c	2019-01-05 21:50:23.367751667 +0100
@@ -26,7 +26,13 @@
  */
 
 /** \cond */
+#ifdef __GLIBC__
 #include <execinfo.h>
+#else
+extern int backtrace (void **__array, int __size);
+extern char **backtrace_symbols (void *const *__array, int __size);
+extern void backtrace_symbols_fd (void *const *__array, int __size, int __fd);
+#endif
 #include <glib-2.0/glib.h>
 #include <stdbool.h>
 #include <stdio.h>
