diff -Nuar Gridcoin-Research-3.6.0.0.orig/src/leveldb/Makefile Gridcoin-Research-3.6.0.0/src/leveldb/Makefile
--- Gridcoin-Research-3.6.0.0.orig/src/leveldb/Makefile	2017-08-21 18:49:22.229840574 +0200
+++ Gridcoin-Research-3.6.0.0/src/leveldb/Makefile	2017-08-21 18:54:21.627824014 +0200
@@ -99,7 +99,7 @@
 
 $(LIBRARY): $(LIBOBJECTS)
 	rm -f $@
-	$(AR) -rs $@ $(LIBOBJECTS)
+	$(AR) $@ $(LIBOBJECTS)
 
 db_bench: db/db_bench.o $(LIBOBJECTS) $(TESTUTIL)
 	$(CXX) $(LDFLAGS) db/db_bench.o $(LIBOBJECTS) $(TESTUTIL) -o $@ $(LIBS)
@@ -178,7 +178,7 @@
 
 $(MEMENVLIBRARY) : $(MEMENVOBJECTS)
 	rm -f $@
-	$(AR) -rs $@ $(MEMENVOBJECTS)
+	$(AR) $@ $(MEMENVOBJECTS)
 
 memenv_test : helpers/memenv/memenv_test.o $(MEMENVLIBRARY) $(LIBRARY) $(TESTHARNESS)
 	$(CXX) $(LDFLAGS) helpers/memenv/memenv_test.o $(MEMENVLIBRARY) $(LIBRARY) $(TESTHARNESS) -o $@ $(LIBS)
