diff -Nuar lld.orig/ELF/Driver.cpp lld/ELF/Driver.cpp
--- a/lld/ELF/Driver.cpp	2022-12-08 23:26:51.815696521 +0100
+++ b/lld/ELF/Driver.cpp	2022-12-08 23:28:27.532581922 +0100
@@ -347,7 +347,7 @@
     if (config->gdbIndex)
       error("-r and --gdb-index may not be used together");
     if (config->icf != ICFLevel::None)
-      error("-r and --icf may not be used together");
+      config->icf = ICFLevel::None;
     if (config->pie)
       error("-r and -pie may not be used together");
     if (config->exportDynamic)
