--- avidemux_2.7.3.orig/avidemux_core/ADM_core/src/ADM_folder_linux.cpp	2019-03-21 18:47:24.000000000 +0100
+++ avidemux_2.7.3/avidemux_core/ADM_core/src/ADM_folder_linux.cpp	2019-04-16 19:48:55.822863485 +0200
@@ -39,7 +39,11 @@
 static std::string canonize(const std::string &in)
 {
     std::string out;
+#ifdef canonicalize_file_name
     char *simple2=canonicalize_file_name(in.c_str());
+#else
+    char *simple2=realpath(in.c_str(), NULL);
+#endif
     if(simple2)
     {
         out=std::string(simple2)+std::string("/");
