diff -Nuar sandbox-2.15.orig/configure.ac sandbox-2.15/configure.ac
--- sandbox-2.15.orig/configure.ac	2019-01-09 14:40:01.000000000 +0100
+++ sandbox-2.15/configure.ac	2019-02-10 02:31:45.202791900 +0100
@@ -354,7 +354,7 @@
 LIBC_PATH=$(AS_IF(
 	dnl GNU linker (bfd & gold)
 	[try_link -Wl,--verbose],
-		[$AWK '/ttempt to open/ { if (($(NF-1) ~ /\/libc\.so/) && ($NF == "succeeded")) LIBC = $(NF-1); }; END {print LIBC}' libctest.log],
+		[$AWK '/ttempt to open/ { if (($(NF-1) ~ /\/libc\.so/) && ($NF == "succeeded")) print $(NF-1); }; /\/libc.so$/ { print $2; }' libctest.log],
 	dnl Solaris linker
 	[try_link -Wl,-m],
 		[set -- `$EGREP -o '/[[^ ]]*/libc.so' libctest.log`; echo $1]
