diff -Nuar llvm.orig/test/ExecutionEngine/RuntimeDyld/X86/COFF_i386.s llvm/test/ExecutionEngine/RuntimeDyld/X86/COFF_i386.s
--- llvm.orig/test/ExecutionEngine/RuntimeDyld/X86/COFF_i386.s	2020-03-23 16:01:02.000000000 +0100
+++ llvm/test/ExecutionEngine/RuntimeDyld/X86/COFF_i386.s	2020-04-28 09:28:15.432288319 +0200
@@ -1,5 +1,6 @@
 // RUN: llvm-mc -triple i686-windows -filetype obj -o %t.obj %s
 // RUN: llvm-rtdyld -triple i686-windows -dummy-extern _printf=0x7ffffffd -dummy-extern _OutputDebugStringA@4=0x7ffffffe -dummy-extern _ExitProcess@4=0x7fffffff -verify -check=%s %t.obj
+// UNSUPPORTED: i686, i386
 
 	.text
 
diff -Nuar llvm.orig/test/ExecutionEngine/RuntimeDyld/X86/MachO_x86-64_PIC_relocations.s llvm/test/ExecutionEngine/RuntimeDyld/X86/MachO_x86-64_PIC_relocations.s
--- llvm.orig/test/ExecutionEngine/RuntimeDyld/X86/MachO_x86-64_PIC_relocations.s	2020-03-23 16:01:02.000000000 +0100
+++ llvm/test/ExecutionEngine/RuntimeDyld/X86/MachO_x86-64_PIC_relocations.s	2020-04-28 09:28:40.686288722 +0200
@@ -1,6 +1,7 @@
 # RUN: rm -rf %t && mkdir -p %t
 # RUN: llvm-mc -triple=x86_64-apple-macosx10.9 -filetype=obj -o %t/test_x86-64.o %s
 # RUN: llvm-rtdyld -triple=x86_64-apple-macosx10.9 -dummy-extern ds1=0xfffffffffffffffe -dummy-extern ds2=0xffffffffffffffff -verify -check=%s %t/test_x86-64.o
+# UNSUPPORTED: i686, i386
 
         .section	__TEXT,__text,regular,pure_instructions
 	.globl	foo
