diff -Nuar rustc-1.31.0-src.orig/src/librustc_target/spec/linux_musl_base.rs rustc-1.31.0-src/src/librustc_target/spec/linux_musl_base.rs
--- rustc-1.31.0-src.orig/src/librustc_target/spec/linux_musl_base.rs	2018-12-16 13:49:44.768376136 +0100
+++ rustc-1.31.0-src/src/librustc_target/spec/linux_musl_base.rs	2018-12-16 13:50:45.375372783 +0100
@@ -8,7 +8,7 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use spec::{LinkerFlavor, TargetOptions};
+use spec::{LinkerFlavor, TargetOptions, RelroLevel};
 
 pub fn opts() -> TargetOptions {
     let mut base = super::linux_base::opts();
@@ -24,5 +24,13 @@
     // These targets allow the user to choose between static and dynamic linking.
     base.crt_static_respected = true;
 
+    // Defaults for dynamic linking
+    base.dynamic_linking = true;
+    base.executables = true;
+    base.has_elf_tls = true;
+    base.has_rpath = true;
+    base.position_independent_executables = true;
+    base.relro_level = RelroLevel::Full;
+
     base
 }
