diff -Nuar dante-1.4.2.orig/osdep.m4 dante-1.4.2/osdep.m4
--- dante-1.4.2.orig/osdep.m4	2019-01-08 10:18:27.177466384 +0100
+++ dante-1.4.2/osdep.m4	2019-01-08 10:18:58.533464650 +0100
@@ -956,7 +956,7 @@
 unset UNIQUEVALS UNIQUESYMBOLS
 if test -s $ERRVALFILE; then
     UNIQUEVALS=`sort $ERRVALFILE | uniq | wc -l | awk '{ print $1 }'`
-    if test $UNIQUEVALS -le 1; then
+    if test $UNIQUEVALS -lt 1; then
         AC_MSG_FAILURE([error: getaddrinfo() error value count too low])
     fi
     UNIQUESYMBOLS=`cat $ERRVALFILE | wc -l | awk '{ print $1 }'`
