diff -Nuar lld.orig/ELF/Driver.cpp lld/ELF/Driver.cpp
--- a/lld/ELF/Driver.cpp	2020-10-07 12:10:48.000000000 +0200
+++ b/lld/ELF/Driver.cpp	2020-10-27 21:40:24.030887913 +0100
@@ -655,10 +655,10 @@
 
 static ICFLevel getICF(opt::InputArgList &args) {
   auto *arg = args.getLastArg(OPT_icf_none, OPT_icf_safe, OPT_icf_all);
-  if (!arg || arg->getOption().getID() == OPT_icf_none)
-    return ICFLevel::None;
-  if (arg->getOption().getID() == OPT_icf_safe)
+  if (!arg || arg->getOption().getID() == OPT_icf_safe)
     return ICFLevel::Safe;
+  if (arg->getOption().getID() == OPT_icf_none)
+    return ICFLevel::None;
   return ICFLevel::All;
 }
 
