diff -Nuar webkitgtk-2.28.4.orig/Source/WebCore/style/StyleResolver.cpp webkitgtk-2.28.4/Source/WebCore/style/StyleResolver.cpp
--- webkitgtk-2.28.4.orig/Source/WebCore/style/StyleResolver.cpp	2020-02-04 11:24:08.000000000 +0100
+++ webkitgtk-2.28.4/Source/WebCore/style/StyleResolver.cpp	2020-11-10 19:50:04.778960961 +0100
@@ -107,7 +107,7 @@
     if (view)
         m_mediaQueryEvaluator = MediaQueryEvaluator { view->mediaType() };
     else
-        m_mediaQueryEvaluator = MediaQueryEvaluator { "all" };
+        m_mediaQueryEvaluator = MediaQueryEvaluator { };
 
     if (root) {
         m_rootDefaultStyle = styleForElement(*root, m_document.renderStyle(), nullptr, RuleMatchingBehavior::MatchOnlyUserAgentRules).renderStyle;
