diff -Nuar work.orig/kernel/nvidia-modeset/nvidia-modeset-linux.c work/kernel/nvidia-modeset/nvidia-modeset-linux.c
--- work.orig/kernel/nvidia-modeset/nvidia-modeset-linux.c	2017-10-26 09:29:25.000000000 +0200
+++ work/kernel/nvidia-modeset/nvidia-modeset-linux.c	2017-11-16 21:01:28.914857489 +0100
@@ -343,8 +343,8 @@
 
 static nvidia_modeset_rm_ops_t __rm_ops = { 0 };
 static nvidia_modeset_callbacks_t nvkms_rm_callbacks = {
-    nvkms_suspend,
-    nvkms_resume
+    .suspend = nvkms_suspend,
+    .resume =  nvkms_resume
 };
 
 static int nvkms_alloc_rm(void)
