diff -Nuar pambase-20150213.orig/Makefile pambase-20150213/Makefile
--- pambase-20150213.orig/Makefile	2019-03-09 21:57:04.745711323 +0100
+++ pambase-20150213/Makefile	2019-03-09 21:57:48.817708885 +0100
@@ -1,7 +1,3 @@
-# Reset this to 'cpp' so it gets traditional syntax; cc -E will not work
-# properly.
-CPP=cpp
-
 # The pam.d file to create
 PAMD=login passwd su system-auth system-login system-local-login system-remote-login system-services other
 
