--- kwin-5.22.4.orig/src/plugins/platforms/x11/common/eglonxbackend.cpp	2021-07-27 12:25:30.000000000 +0200
+++ kwin-5.22.4/src/plugins/platforms/x11/common/eglonxbackend.cpp	2021-08-06 10:55:04.330362790 +0200
@@ -224,7 +224,7 @@
         //       cannot be an xcb_window_t.
         surface = eglCreatePlatformWindowSurfaceEXT(eglDisplay(), config(), (void *) &window, nullptr);
     } else {
-        surface = eglCreateWindowSurface(eglDisplay(), config(), window, nullptr);
+        surface = eglCreateWindowSurface(eglDisplay(), config(), &window, nullptr);
     }

     return surface;
