diff -Nuar cfe-8.0.0.src.orig/lib/Driver/ToolChains/Linux.cpp cfe-8.0.0.src/lib/Driver/ToolChains/Linux.cpp
--- cfe-8.0.0.src.orig/lib/Driver/ToolChains/Linux.cpp	2019-03-27 10:51:10.426081551 +0100
+++ cfe-8.0.0.src/lib/Driver/ToolChains/Linux.cpp	2019-03-27 10:53:32.457073695 +0100
@@ -979,6 +979,13 @@
   return Generic_ELF::IsMathErrnoDefault();
 }
 
+unsigned Linux::GetDefaultStackProtectorLevel(bool KernelOrKext) const {
+  StringRef VendorName = Linux::getTriple().getVendorName();
+  if (VendorName.compare("gentoo") == 0)
+    return 2;
+  return 1;
+}
+
 SanitizerMask Linux::getSupportedSanitizers() const {
   const bool IsX86 = getTriple().getArch() == llvm::Triple::x86;
   const bool IsX86_64 = getTriple().getArch() == llvm::Triple::x86_64;
diff -Nuar cfe-8.0.0.src.orig/lib/Driver/ToolChains/Linux.h cfe-8.0.0.src/lib/Driver/ToolChains/Linux.h
--- cfe-8.0.0.src.orig/lib/Driver/ToolChains/Linux.h	2018-11-05 21:57:46.000000000 +0100
+++ cfe-8.0.0.src/lib/Driver/ToolChains/Linux.h	2019-03-27 10:54:14.311071380 +0100
@@ -39,6 +39,7 @@
                            llvm::opt::ArgStringList &CC1Args) const override;
   CXXStdlibType GetDefaultCXXStdlibType() const override;
   bool isPIEDefault() const override;
+  unsigned GetDefaultStackProtectorLevel(bool KernelOrKext) const override;
   bool IsMathErrnoDefault() const override;
   SanitizerMask getSupportedSanitizers() const override;
   void addProfileRTLibs(const llvm::opt::ArgList &Args,
diff -Nuar cfe-8.0.0.src.orig/test/Driver/fsanitize.c cfe-8.0.0.src/test/Driver/fsanitize.c
--- cfe-8.0.0.src.orig/test/Driver/fsanitize.c	2019-01-12 12:18:35.000000000 +0100
+++ cfe-8.0.0.src/test/Driver/fsanitize.c	2019-03-27 10:53:32.458073695 +0100
@@ -687,12 +687,12 @@
 // RUN: %clang -fno-sanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=NOSP
 // NOSP-NOT: "-fsanitize=safe-stack"
 
-// RUN: %clang -target x86_64-linux-gnu -fsanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=NO-SP
+// RUN: %clang -target x86_64-linux-gnu -fsanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=SP
 // RUN: %clang -target x86_64-linux-gnu -fsanitize=address,safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=SP-ASAN
 // RUN: %clang -target x86_64-linux-gnu -fstack-protector -fsanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=SP
 // RUN: %clang -target x86_64-linux-gnu -fsanitize=safe-stack -fstack-protector-all -### %s 2>&1 | FileCheck %s -check-prefix=SP
-// RUN: %clang -target arm-linux-androideabi -fsanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=NO-SP
-// RUN: %clang -target aarch64-linux-android -fsanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=NO-SP
+// RUN: %clang -target arm-linux-androideabi -fsanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=SP
+// RUN: %clang -target aarch64-linux-android -fsanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=SP
 // RUN: %clang -target i386-contiki-unknown -fsanitize=safe-stack -### %s 2>&1 | FileCheck %s -check-prefix=NO-SP
 // NO-SP-NOT: stack-protector
 // NO-SP: "-fsanitize=safe-stack"
diff -Nuar cfe-8.0.0.src.orig/test/Driver/stack-protector.c cfe-8.0.0.src/test/Driver/stack-protector.c
--- cfe-8.0.0.src.orig/test/Driver/stack-protector.c	2017-09-06 01:50:58.000000000 +0200
+++ cfe-8.0.0.src/test/Driver/stack-protector.c	2019-03-27 10:53:32.458073695 +0100
@@ -24,6 +24,20 @@
 // SSP-ALL: "-stack-protector" "3"
 // SSP-ALL-NOT: "-stack-protector-buffer-size" 
 
+// RUN: %clang -target x86_64-gentoo-linux-musl -### %s 2>&1 | FileCheck %s -check-prefix=GENTOO
+// GENTOO: "-stack-protector" "2"
+
+// RUN: %clang -target x86_64-gentoo-linux-musl -fstack-protector -### %s 2>&1 | FileCheck %s -check-prefix=GENTOO_SPS
+// GENTOO_SPS: "-stack-protector" "2"
+
+// RUN: %clang -target x86_64-gentoo-linux-musl -fstack-protector-all -### %s 2>&1 | FileCheck %s -check-prefix=GENTOO_ALL
+// GENTOO_ALL: "-stack-protector" "3"
+// GENTOO_ALL-NOT: "-stack-protector-buffer-size"
+
+// RUN: %clang -target x86_64-gentoo-linux-musl -fno-stack-protector -### %s 2>&1 | FileCheck %s -check-prefix=GENTOO_NOSSP
+// GENTOO_NOSSP-NOT: "-stack-protector"
+// GENTOO_NOSSP-NOT: "-stack-protector-buffer-size"
+
 // RUN: %clang -target x86_64-scei-ps4 -### %s 2>&1 | FileCheck %s -check-prefix=SSP-PS4
 // RUN: %clang -target x86_64-scei-ps4 -fstack-protector -### %s 2>&1 | FileCheck %s -check-prefix=SSP-PS4
 // SSP-PS4: "-stack-protector" "2"
