diff -Nuar nfs-utils-2.3.3.orig/configure.ac nfs-utils-2.3.3/configure.ac
--- nfs-utils-2.3.3.orig/configure.ac	2019-01-16 10:54:26.012328110 +0100
+++ nfs-utils-2.3.3/configure.ac	2019-01-16 10:55:00.743326189 +0100
@@ -408,7 +408,7 @@
 fi
 
 dnl libdnsidmap specific checks
-AC_CHECK_LIB([resolv], [__res_querydomain], , AC_MSG_ERROR(res_querydomain needed))
+AC_CHECK_LIB([resolv], [res_querydomain], , AC_MSG_ERROR(res_querydomain needed))
 
 AC_ARG_ENABLE([ldap],
 	[AS_HELP_STRING([--disable-ldap],[Disable support for LDAP @<:default=detect@:>@])])
