diff -Nuar boinc-client_release-7.8-7.8.4.orig/lib/crypt.h boinc-client_release-7.8-7.8.4/lib/crypt.h
--- boinc-client_release-7.8-7.8.4.orig/lib/crypt.h	2017-11-10 22:54:05.000000000 +0100
+++ boinc-client_release-7.8-7.8.4/lib/crypt.h	2018-01-07 19:34:58.427327916 +0100
@@ -26,7 +26,7 @@
 
 #include <openssl/rsa.h>
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) /* OpenSSL 1.1.0+ */
+#if (defined(OPENSSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)) && !defined(LIBRESSL_VERSION_NUMBER)/* OpenSSL 1.1.0+ or not LibreSSL */
 #define HAVE_OPAQUE_EVP_PKEY 1 /* since 1.1.0 -pre3 */
 #define HAVE_OPAQUE_RSA_DSA_DH 1 /* since 1.1.0 -pre5 */
 #endif
