diff -Nuar samba-4.5.3.orig/source3/libsmb/wscript samba-4.5.3/source3/libsmb/wscript
--- samba-4.5.3.orig/source3/libsmb/wscript	2016-08-11 09:51:04.000000000 +0200
+++ samba-4.5.3/source3/libsmb/wscript	2017-01-07 21:12:17.218808409 +0100
@@ -19,7 +19,7 @@
                               libsmb_setget.c''',
                        public_deps='''
                                    talloc
-                                   param
+                                   smbconf
                                    libsmb
                                    KRBCLIENT
                                    msrpc3
diff -Nuar samba-4.5.3.orig/source3/param/wscript_build samba-4.5.3/source3/param/wscript_build
--- samba-4.5.3.orig/source3/param/wscript_build	2017-01-07 21:11:52.146872128 +0100
+++ samba-4.5.3/source3/param/wscript_build	2017-01-07 21:12:17.218808409 +0100
@@ -6,7 +6,7 @@
 
 bld.SAMBA3_SUBSYSTEM('LOADPARM_CTX',
                      source='loadparm_ctx.c',
-                     deps='''talloc param''')
+                     deps='''talloc smbconf''')
 
 bld.SAMBA_GENERATOR('s3_param_proto_h',
                     source= '../../script/generate_param.py ../../docs-xml/smbdotconf/parameters.all.xml',
@@ -17,18 +17,18 @@
 if not bld.env.disable_python:
     bld.SAMBA3_PYTHON('pys3param',
                   source='pyparam.c',
-                  deps='param',
+                  deps='smbconf',
                   public_deps='samba-hostconfig pytalloc-util talloc',
                   realname='samba/samba3/param.so')
 
 bld.SAMBA3_SUBSYSTEM('param_service',
                      source='service.c',
-                     deps = 'USER_UTIL param PRINTING')
+                     deps = 'USER_UTIL smbconf PRINTING')
 
 bld.SAMBA3_BINARY('test_lp_load',
                  source='test_lp_load.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  popt_samba3''',
                  install=False)
diff -Nuar samba-4.5.3.orig/source3/smbd/notifyd/wscript_build samba-4.5.3/source3/smbd/notifyd/wscript_build
--- samba-4.5.3.orig/source3/smbd/notifyd/wscript_build	2016-08-11 09:51:04.000000000 +0200
+++ samba-4.5.3/source3/smbd/notifyd/wscript_build	2017-01-07 21:12:17.218808409 +0100
@@ -8,12 +8,12 @@
                   source='tests.c',
                   install=False,
                   deps='''
-                    param
+                       smbconf
                   ''')
 
 bld.SAMBA3_BINARY('notifydd',
                   source='notifydd.c',
                   install=False,
                   deps='''notifyd
-                    param
+                       smbconf
                   ''')
diff -Nuar samba-4.5.3.orig/source3/winbindd/wscript_build samba-4.5.3/source3/winbindd/wscript_build
--- samba-4.5.3.orig/source3/winbindd/wscript_build	2016-08-11 09:51:04.000000000 +0200
+++ samba-4.5.3/source3/winbindd/wscript_build	2017-01-07 21:12:17.218808409 +0100
@@ -116,7 +116,7 @@
 
 bld.SAMBA3_LIBRARY('nss_info',
                    source='nss_info.c',
-                   deps='samba-util param',
+                   deps='samba-util smbconf',
                    private_library=True)
 
 bld.SAMBA3_MODULE('nss_info_template',
diff -Nuar samba-4.5.3.orig/source3/wscript_build samba-4.5.3/source3/wscript_build
--- samba-4.5.3.orig/source3/wscript_build	2017-01-07 21:11:52.389871510 +0100
+++ samba-4.5.3/source3/wscript_build	2017-01-07 21:12:17.219808407 +0100
@@ -57,7 +57,7 @@
 
 bld.SAMBA3_LIBRARY('gse',
                    source='librpc/crypto/gse_krb5.c librpc/crypto/gse.c',
-                   deps='krb5samba gensec param KRBCLIENT secrets3',
+                   deps='krb5samba gensec smbconf KRBCLIENT secrets3',
                    private_library=True)
 
 bld.SAMBA3_LIBRARY('msrpc3',
@@ -167,22 +167,6 @@
                      source='lib/server_mutex.c',
                      deps='talloc')
 
-bld.SAMBA3_SUBSYSTEM('param',
-                   source='''param/loadparm.c
-                   lib/sharesec.c
-                   lib/ldap_debug_handler.c
-                   lib/util_names.c''',
-                   deps='''samba-util
-                   PARAM_UTIL
-                   ldap
-                   lber
-                   LOADPARM_CTX
-                   samba3core
-                   smbconf
-                   param_local.h
-                   param_global.h
-                   cups''')
-
 # this includes only the low level parse code, not stuff
 # that requires knowledge of security contexts
 bld.SAMBA3_SUBSYSTEM('REG_PARSE_PRS',
@@ -196,43 +180,61 @@
                     source='registry/reg_api_regf.c',
                     deps='samba-util')
 
-bld.SAMBA3_LIBRARY('smbregistry',
-                   source='''registry/reg_api.c
-                   registry/reg_dispatcher.c
-                   registry/reg_cachehook.c
-                   registry/reg_objects.c
-                   registry/reg_util_internal.c
-                   lib/util_nttoken.c
-                   registry/reg_backend_db.c
-                   registry/reg_parse_internal.c
-                   lib/cbuf.c
-                   lib/srprs.c
-                   registry/reg_init_basic.c''',
-                   deps='''smbd_shim tdb-wrap3 NDR_SECURITY util_tdb talloc
-                   replace util_reg samba-util samba-security
-                   errors3 dbwrap samba3-util''',
-                   allow_undefined_symbols=True,
-                   private_library=True)
+# Do not link against this use 'smbconf'
+bld.SAMBA3_SUBSYSTEM('SMBREGISTRY',
+                   source='''
+                          registry/reg_api.c
+                          registry/reg_dispatcher.c
+                          registry/reg_cachehook.c
+                          registry/reg_objects.c
+                          registry/reg_util_internal.c
+                          lib/util_nttoken.c
+                          registry/reg_backend_db.c
+                          registry/reg_parse_internal.c
+                          lib/srprs.c
+                          registry/reg_init_basic.c
+                          ''',
+                   deps='''
+                        smbd_shim
+                        tdb-wrap3
+                        NDR_SECURITY
+                        util_tdb
+                        talloc
+                        replace
+                        util_reg
+                        samba-util
+                        samba-security
+                        errors3
+                        dbwrap
+                        samba3-util
+                        ''')
 
+# Do not link against this use 'smbconf'
 bld.SAMBA3_SUBSYSTEM('REG_SMBCONF',
-                    source='''registry/reg_backend_smbconf.c
-                    registry/reg_init_smbconf.c
-                    registry/reg_util_token.c
-                    registry/reg_api_util.c''',
-                    deps='smbregistry')
+                    source='''
+                           registry/reg_backend_smbconf.c
+                           registry/reg_init_smbconf.c
+                           registry/reg_util_token.c
+                           registry/reg_api_util.c
+                           ''',
+                    deps='SMBREGISTRY')
 
 bld.SAMBA3_SUBSYSTEM('REG_FULL',
-                    source='''registry/reg_backend_printing.c
-                    registry/reg_backend_shares.c
-                    registry/reg_backend_netlogon_params.c
-                    registry/reg_backend_prod_options.c
-                    registry/reg_backend_tcpip_params.c
-                    registry/reg_backend_hkpt_params.c
-                    registry/reg_backend_current_version.c
-                    registry/reg_backend_perflib.c
-                    registry/reg_init_full.c
-                    registry/reg_perfcount.c''',
-                    deps='REG_SMBCONF tdb-wrap3')
+                    source='''
+                           registry/reg_backend_printing.c
+                           registry/reg_backend_shares.c
+                           registry/reg_backend_netlogon_params.c
+                           registry/reg_backend_prod_options.c
+                           registry/reg_backend_tcpip_params.c
+                           registry/reg_backend_hkpt_params.c
+                           registry/reg_backend_current_version.c
+                           registry/reg_backend_perflib.c
+                           registry/reg_init_full.c
+                           registry/reg_perfcount.c''',
+                    deps='''
+                         smbconf
+                         tdb-wrap3
+                         ''')
 
 bld.SAMBA3_LIBRARY('popt_samba3',
                    source='lib/popt_common.c',
@@ -249,22 +251,34 @@
                      public_deps='krb5samba k5crypto gssapi LIBTSOCKET CLDAP LIBNMB')
 
 bld.SAMBA3_SUBSYSTEM('samba3util',
-                   source='''lib/system.c
-                   lib/sendfile.c
-                   lib/recvfile.c
-                   lib/time.c
-                   lib/util_sid.c
-                   lib/util_specialsids.c
-                   lib/util_file.c
-                   lib/util.c
-                   lib/util_path.c
-                   lib/util_procid.c
-                   lib/util_sock.c
-                   lib/util_tsock.c
-                   lib/util_transfer_file.c
-                   lib/sys_popen.c
-                   lib/sock_exec.c''',
-                   deps='ndr LIBTSOCKET samba-security NDR_SECURITY samba-util util_tdb sys_rw iov_buf')
+                   source='''
+                          lib/system.c
+                          lib/sendfile.c
+                          lib/recvfile.c
+                          lib/time.c
+                          lib/util_sid.c
+                          lib/util_specialsids.c
+                          lib/util_file.c
+                          lib/util.c
+                          lib/util_path.c
+                          lib/util_procid.c
+                          lib/util_sock.c
+                          lib/util_tsock.c
+                          lib/util_transfer_file.c
+                          lib/sys_popen.c
+                          lib/sock_exec.c
+                          ''',
+                   deps='''
+                        CHARSET3
+                        ndr
+                        LIBTSOCKET
+                        samba-security
+                        NDR_SECURITY
+                        samba-util
+                        util_tdb
+                        sys_rw
+                        iov_buf
+                        ''')
 
 if bld.env.with_ctdb:
     SAMBA_CLUSTER_SUPPORT_SOURCES='''
@@ -358,7 +372,7 @@
                         UNIX_MSG
                         POLL_FUNCS_TEVENT
                         interfaces
-                        param
+                        smbconf
                         dbwrap
                         samba3-util
                         errors3
@@ -423,16 +437,23 @@
 # PLEASE DO NOT make it depend on high level libraries like PDB, if you are
 # doing that your design is wrong and needs changing. -SSS
 bld.SAMBA3_LIBRARY('secrets3',
-                   source='''passdb/secrets.c
-                   passdb/machine_account_secrets.c
-                   passdb/machine_sid.c
-                   passdb/secrets_lsa.c''',
-                   deps='NDR_SECRETS param samba3util dbwrap',
+                   source='''
+                          passdb/secrets.c
+                          passdb/machine_account_secrets.c
+                          passdb/machine_sid.c
+                          passdb/secrets_lsa.c
+                          ''',
+                   deps='''
+                        NDR_SECRETS
+                        smbconf
+                        samba3util
+                        dbwrap
+                        ''',
                    private_library=True)
 
 bld.SAMBA3_LIBRARY('smbldap',
                     source='lib/smbldap.c',
-                    deps='ldap lber samba-util param',
+                    deps='ldap lber samba-util smbconf',
                     enabled=bld.CONFIG_SET("HAVE_LDAP"),
                     private_library=False,
                     abi_directory='lib/ABI',
@@ -442,20 +463,33 @@
                     public_headers='include/smbldap.h include/smb_ldap.h')
 
 bld.SAMBA3_LIBRARY('ads',
-                   source='''libads/ldap.c
-                   libads/sasl.c
-                   libads/sasl_wrapping.c
-                   libads/krb5_setpw.c
-                   libads/kerberos_util.c
-                   libads/ldap_user.c
-                   libads/ads_struct.c
-                   libads/kerberos_keytab.c
-                   libads/disp_sec.c
-                   libads/ldap_utils.c
-                   libads/ldap_schema.c
-                   libads/util.c
-                   libads/ndr.c''',
-                   deps='cli-ldap-common krb5samba ldap lber KRBCLIENT param LIBNMB libsmb DCUTIL smbldap',
+                   source='''
+                          libads/ldap.c
+                          libads/sasl.c
+                          libads/sasl_wrapping.c
+                          libads/krb5_setpw.c
+                          libads/kerberos_util.c
+                          libads/ldap_user.c
+                          libads/ads_struct.c
+                          libads/kerberos_keytab.c
+                          libads/disp_sec.c
+                          libads/ldap_utils.c
+                          libads/ldap_schema.c
+                          libads/util.c
+                          libads/ndr.c
+                          ''',
+                   deps='''
+                        cli-ldap-common
+                        krb5samba
+                        ldap
+                        lber
+                        KRBCLIENT
+                        smbconf
+                        LIBNMB
+                        libsmb
+                        DCUTIL
+                        smbldap
+                        ''',
                    private_library=True)
 
 bld.SAMBA3_SUBSYSTEM('LIBADS_SERVER',
@@ -466,23 +500,44 @@
                     source='libads/ldap_printer.c',
                     deps='samba-util krb5samba')
 
+# Only smbconf should link against this subsystem, else we will create
+# a circular dependency.
+bld.SAMBA3_SUBSYSTEM('SMBCONF_PARAM',
+                   source='''
+                          param/loadparm.c
+                          lib/sharesec.c
+                          lib/ldap_debug_handler.c
+                          lib/util_names.c
+                          ''',
+                   deps='''
+                        samba-util
+                        PARAM_UTIL
+                        ldap
+                        lber
+                        LOADPARM_CTX
+                        samba3core
+                        param_local.h
+                        param_global.h
+                        cups
+                        ''')
+
 bld.SAMBA3_LIBRARY('smbconf',
                    source='''lib/smbconf/smbconf_init.c
                    lib/smbconf/smbconf_reg.c''',
                    deps='''
-                   CHARSET3
-                   LIBSMBCONF
-                   REG_SMBCONF
-                   SAMBA_VERSION
-                   cap
-                   charset
-                   cli_smb_common
-                   errors3
-                   param
-                   samba-util
-                   smbregistry
-                   talloc
-                   util_reg''',
+                        CHARSET3
+                        LIBSMBCONF
+                        REG_SMBCONF
+                        SAMBA_VERSION
+                        cap
+                        charset
+                        cli_smb_common
+                        errors3
+                        SMBCONF_PARAM
+                        samba-util
+                        talloc
+                        util_reg
+                        ''',
                    public_headers='../lib/smbconf/smbconf.h',
                    pc_files=[],
                    vnum='0')
@@ -615,8 +670,7 @@
                    vfs
                    vfs_default
                    vfs_posixacl
-		   inotify
-                   popt_samba3
+                   inotify
                    samba3core
                    smbd_conn
                    param_service
@@ -680,8 +734,18 @@
                     deps='PRINTBASE LIBADS_PRINTER tdb printing_migrate')
 
 bld.SAMBA3_LIBRARY('printing_migrate',
-                    source='printing/nt_printing_migrate.c rpc_client/cli_winreg_spoolss.c printing/nt_printing_os2.c',
-                    deps='NDR_NTPRINTING cli_spoolss RPC_NDR_WINREG LIBCLI_WINREG param',
+                    source='''
+                           printing/nt_printing_migrate.c
+                           rpc_client/cli_winreg_spoolss.c
+                           printing/nt_printing_os2.c
+                           ''',
+                    deps='''
+                         NDR_NTPRINTING
+                         cli_spoolss
+                         RPC_NDR_WINREG
+                         LIBCLI_WINREG
+                         smbconf
+                         ''',
                     private_library=True)
 
 bld.SAMBA3_SUBSYSTEM('PRINTING',
@@ -786,8 +850,19 @@
                     deps='talloc samba3-util')
 
 bld.SAMBA3_LIBRARY('samba3-util',
-                   source='''lib/util_sec.c lib/util_str.c lib/adt_tree.c lib/util_malloc.c lib/namearray.c lib/file_id.c''',
-                   deps='samba-util charset',
+                   source='''
+                          lib/util_sec.c
+                          lib/util_str.c
+                          lib/adt_tree.c
+                          lib/util_malloc.c
+                          lib/namearray.c
+                          lib/file_id.c
+                          lib/cbuf.c
+                          ''',
+                   deps='''
+                        samba-util
+                        charset
+                        ''',
                    private_library=True)
 
 bld.SAMBA_LIBRARY('xattr_tdb',
@@ -815,14 +890,28 @@
                    private_library=True)
 
 bld.SAMBA3_LIBRARY('libcli_netlogon3',
-                   source='rpc_client/cli_netlogon.c rpc_client/util_netlogon.c',
-                   deps='msrpc3 RPC_NDR_NETLOGON INIT_NETLOGON cliauth param NETLOGON_CREDS_CLI',
+                   source='''
+                          rpc_client/cli_netlogon.c
+                          rpc_client/util_netlogon.c
+                          ''',
+                   deps='''
+                        msrpc3
+                        RPC_NDR_NETLOGON
+                        INIT_NETLOGON
+                        cliauth
+                        smbconf
+                        NETLOGON_CREDS_CLI''',
                    private_library=True)
 
 bld.SAMBA3_LIBRARY('cli_spoolss',
-                   source='''rpc_client/cli_spoolss.c
-                   rpc_client/init_spoolss.c''',
-                   deps='RPC_NDR_SPOOLSS param secrets3',
+                   source='''
+                          rpc_client/cli_spoolss.c
+                          rpc_client/init_spoolss.c
+                          ''',
+                   deps='''
+                        RPC_NDR_SPOOLSS
+                        smbconf
+                        secrets3''',
                    private_library=True)
 
 bld.SAMBA3_SUBSYSTEM('LIBCLI_WINREG',
@@ -856,7 +945,7 @@
 
 bld.SAMBA3_BINARY('smbd/smbd',
                  source='smbd/server.c smbd/smbd_cleanupd.c',
-                 deps='smbd_base EPMD LSASD FSSD MDSSD',
+                 deps='popt_samba3 smbd_base EPMD LSASD FSSD MDSSD',
                  install_path='${SBINDIR}')
 
 bld.SAMBA3_BINARY('nmbd/nmbd',
@@ -873,12 +962,12 @@
                  nmbd/nmbd_subnetdb.c nmbd/nmbd_winsproxy.c nmbd/nmbd_winsserver.c
                  nmbd/nmbd_workgroupdb.c nmbd/nmbd_synclists.c''',
                  deps='''
-                 talloc
-                 tevent
-                 param
-                 libsmb
-                 popt_samba3
-                 ''',
+                      talloc
+                      tevent
+                      smbconf
+                      libsmb
+                      popt_samba3
+                      ''',
                  install_path='${SBINDIR}')
 
 
@@ -1025,7 +1114,7 @@
                  popt_samba3
                  pdb
                  libsmb
-                 param
+                 smbconf
                  ndr-standard
                  msrpc3
                  SMBREADLINE
@@ -1056,17 +1145,17 @@
                  client/clitar.c
                  client/dnsbrowse.c''',
                  deps='''
-                 talloc
-                 popt_samba3
-                 param
-                 ndr-standard
-                 SMBREADLINE
-                 libsmb
-                 msrpc3
-                 RPC_NDR_SRVSVC
-                 cli_smb_common
-		 archive
-                 ''')
+                      talloc
+                      popt_samba3
+                      smbconf
+                      ndr-standard
+                      SMBREADLINE
+                      libsmb
+                      msrpc3
+                      RPC_NDR_SRVSVC
+                      cli_smb_common
+                      archive
+                      ''')
 
 bld.SAMBA3_BINARY('net',
                  source='''utils/net.c
@@ -1127,7 +1216,7 @@
                  popt_samba3
                  pdb
                  libsmb
-                 param
+                 smbconf
                  KRBCLIENT
                  ndr-standard
                  msrpc3
@@ -1163,17 +1252,18 @@
                  deps='''
                  talloc
                  popt_samba3
-                 param
+                 smbconf
                  REGFIO''')
 
 bld.SAMBA3_BINARY('smbspool',
                  source='client/smbspool.c',
                  deps='''
-                 talloc
-                 popt_samba3
-                 param
-                 libsmb
-                 samba3core''')
+                      talloc
+                      popt_samba3
+                      smbconf
+                      libsmb
+                      samba3core
+                      ''')
 
 bld.SAMBA3_BINARY('smbspool_krb5_wrapper',
                  source='client/smbspool_krb5_wrapper.c',
@@ -1187,9 +1277,11 @@
 bld.SAMBA3_BINARY('testparm',
                  source='utils/testparm.c',
                  deps='''
-                 talloc
-                 param
-                 popt_samba3''')
+                      talloc
+                      popt_samba3
+                      smbconf
+                      ''',
+                 install=False)
 
 smbstatus_source = 'utils/status.c smbd/notify_msg.c'
 
@@ -1201,18 +1293,19 @@
 bld.SAMBA3_BINARY('smbstatus',
                  source=smbstatus_source,
                  deps='''
-                 talloc
-                 param
-                 popt_samba3
-                 smbd_base
-                 LOCKING
-                 PROFILE''')
+                      talloc
+                      smbconf
+                      popt_samba3
+                      smbd_base
+                      LOCKING
+                      PROFILE
+                      ''')
 
 bld.SAMBA3_BINARY('smbcontrol',
                  source='utils/smbcontrol.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  popt_samba3
                  PRINTBASE''')
 
@@ -1220,7 +1313,7 @@
                  source='utils/smbtree.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  libsmb
                  msrpc3
                  popt_samba3
@@ -1230,7 +1323,7 @@
                  source='utils/smbpasswd.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  pdb
                  PASSWD_UTIL
                  PASSCHANGE''')
@@ -1239,7 +1332,7 @@
                  source='utils/pdbedit.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  popt_samba3
                  pdb
                  PASSWD_UTIL''')
@@ -1255,7 +1348,7 @@
                  source='utils/nmblookup.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  popt_samba3
                  LIBNMB''')
 
@@ -1291,28 +1384,29 @@
                  torture/bench_pthreadpool.c
                  torture/wbc_async.c''',
                  deps='''
-                 talloc
-                 param
-                 libsmb
-                 msrpc3
-                 TLDAP
-                 RPC_NDR_ECHO
-                 WB_REQTRANS
-                 LOCKING
-                 NDR_OPEN_FILES
-                 idmap
-                 IDMAP_TDB_COMMON
-                 samba-cluster-support
-                 ''',
+                      talloc
+                      smbconf
+                      libsmb
+                      msrpc3
+                      TLDAP
+                      RPC_NDR_ECHO
+                      WB_REQTRANS
+                      LOCKING
+                      NDR_OPEN_FILES
+                      idmap
+                      IDMAP_TDB_COMMON
+                      samba-cluster-support
+                      ''',
                  cflags='-DWINBINDD_SOCKET_DIR=\"%s\"' % bld.env.WINBINDD_SOCKET_DIR,
                  install=False)
 
 bld.SAMBA3_BINARY('smbconftort',
                  source='lib/smbconf/testsuite.c',
                  deps='''
-                 talloc
-                 param
-                 popt_samba3''',
+                      talloc
+                      smbconf
+                      popt_samba3
+                      ''',
                  install=False)
 
 bld.SAMBA3_BINARY('replacetort',
@@ -1323,22 +1417,25 @@
 bld.SAMBA3_BINARY('msgtest',
                  source='torture/msgtest.c',
                  deps='''
-                 talloc
-                 param''',
+                      talloc
+                      smbconf
+                      ''',
                  install=False)
 
 bld.SAMBA3_BINARY('msg_sink',
                  source='torture/msg_sink.c',
                  deps='''
-                 talloc
-                 param''',
+                      talloc
+                      smbconf
+                      ''',
                  install=False)
 
 bld.SAMBA3_BINARY('msg_source',
                  source='torture/msg_source.c',
                  deps='''
-                 talloc
-                 param''',
+                      talloc
+                      smbconf
+                      ''',
                  install=False)
 
 bld.SAMBA3_BINARY('smbcacls',
@@ -1363,7 +1460,7 @@
                  source='utils/eventlogadm.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  LIBEVENTLOG''')
 
 bld.SAMBA3_BINARY('sharesec',
@@ -1389,9 +1486,11 @@
                  torture/vfstest.c
                  torture/vfstest_chain.c''',
                  deps='''
-                 vfs
-                 popt_samba3
-                 SMBREADLINE''',
+                      vfs
+                      popt_samba3
+                      smbconf
+                      SMBREADLINE
+                      ''',
                  install=False)
 
 bld.SAMBA3_BINARY('log2pcap',
@@ -1403,7 +1502,7 @@
                  source='torture/locktest2.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  libsmb
                  LOCKING''',
                  install=False)
@@ -1417,15 +1516,16 @@
                  source='utils/smbfilter.c',
                  deps='''
                  talloc
-                 param
+                 smbconf
                  LIBNMB''',
                  install=False)
 
 bld.SAMBA3_BINARY('versiontest',
                  source='lib/version_test.c',
                  deps='''
-                 SAMBA_VERSION
-                 param''',
+                      SAMBA_VERSION
+                      smbconf
+                      ''',
                  install=False)
 
 bld.SAMBA3_BINARY('ntlm_auth',
@@ -1472,8 +1572,9 @@
 bld.SAMBA3_BINARY('vlp',
                  source='printing/tests/vlp.c',
                  deps='''
-                 talloc
-                 param''',
+                      talloc
+                      smbconf
+                      ''',
                  install=False)
 
 if not bld.env.disable_python:
@@ -1494,7 +1595,7 @@
                             utils/regedit_wrap.c utils/regedit_treeview.c
                             utils/regedit_valuelist.c utils/regedit_dialog.c
                             utils/regedit_hexedit.c utils/regedit_list.c""",
-                  deps='ncurses menu panel form registry param popt_samba3 smbregistry',
+                  deps='ncurses menu panel form registry smbconf popt_samba3',
                   enabled=bld.env.build_regedit)
 
 bld.SAMBA3_BINARY('spotlight2sparql',
diff -Nuar samba-4.5.3.orig/source4/dsdb/wscript_build samba-4.5.3/source4/dsdb/wscript_build
--- samba-4.5.3.orig/source4/dsdb/wscript_build	2017-01-07 21:11:52.184872031 +0100
+++ samba-4.5.3/source4/dsdb/wscript_build	2017-01-07 21:12:17.218808409 +0100
@@ -39,7 +39,7 @@
 
 bld.SAMBA_LIBRARY('dsdb_garbage_collect_tombstones',
                   source='kcc/garbage_collect_tombstones.c',
-                  deps='samdb param RPC_NDR_DRSUAPI',
+                  deps='samdb RPC_NDR_DRSUAPI',
                   private_library=True)
 
 bld.SAMBA_MODULE('service_kcc',
