diff -Nuar libixion-0.9.1.orig/configure libixion-0.9.1/configure
--- libixion-0.9.1.orig/configure	2016-03-13 01:41:11.780038808 +0100
+++ libixion-0.9.1/configure	2016-03-13 01:41:49.089284206 +0100
@@ -17476,7 +17476,7 @@
 fi
 
 
-CPPFLAGS="$CPPFLAGS -g -Os -fvisibility=hidden"
+CPPFLAGS="$CPPFLAGS -fvisibility=hidden"
 
 ac_ext=cpp
 ac_cpp='$CXXCPP $CPPFLAGS'
diff -Nuar libixion-0.9.1.orig/configure.ac libixion-0.9.1/configure.ac
--- libixion-0.9.1.orig/configure.ac	2016-03-13 01:41:06.769005849 +0100
+++ libixion-0.9.1/configure.ac	2016-03-13 01:41:42.702242196 +0100
@@ -92,7 +92,7 @@
 ])
 AM_CONDITIONAL([BUILD_PYTHON], [test "x$enable_python" != "xno"])
 
-CPPFLAGS="$CPPFLAGS -g -Os -fvisibility=hidden"
+CPPFLAGS="$CPPFLAGS -fvisibility=hidden"
 
 BOOST_THREADS
 BOOST_PROGRAM_OPTIONS
