--- wxWidgets-3.0.5.1.orig/configure	2023-05-07 17:58:28.782795223 +0200
+++ wxWidgets-3.0.5.1/configure	2023-05-07 18:02:58.203631708 +0200
@@ -22300,6 +22300,7 @@
 
                         #include <zlib.h>
                         #include <stdio.h>
+                        #include <stdlib.h>
 
                         int main()
                         {
@@ -22422,6 +22423,7 @@
 
                     #include <png.h>
                     #include <stdio.h>
+                    #include <stdlib.h>
 
                     int main()
                     {
