--- webkitgtk-2.4.11.orig/Source/WebKit2/Shared/API/c/WKString.h	2016-04-10 08:48:37.000000000 +0200
+++ webkitgtk-2.4.11/Source/WebKit2/Shared/API/c/WKString.h	2017-12-20 22:14:19.097225121 +0100
@@ -28,6 +28,7 @@

 #include <WebKit2/WKBase.h>
 #include <stddef.h>
+#include <uchar.h>

 #ifndef __cplusplus
 #include <stdbool.h>
@@ -39,7 +40,7 @@

 #if !defined(WIN32) && !defined(_WIN32) \
     && !((defined(__CC_ARM) || defined(__ARMCC__)) && !defined(__linux__)) /* RVCT */
-    typedef unsigned short WKChar;
+    typedef char16_t WKChar;
 #else
     typedef wchar_t WKChar;
 #endif
