diff -Nuar exiv2-0.27.0-Source.orig/src/futils.cpp exiv2-0.27.0-Source/src/futils.cpp
--- exiv2-0.27.0-Source.orig/src/futils.cpp	2018-12-20 14:45:16.000000000 +0100
+++ exiv2-0.27.0-Source/src/futils.cpp	2019-03-10 23:13:12.503388861 +0100
@@ -53,6 +53,10 @@
 #include <algorithm>
 #include <stdexcept>
 
+#ifndef __GLIBC__
+#undef _GNU_SOURCE
+#endif
+
 #ifdef EXV_HAVE_STRERROR_R
 #ifdef _GNU_SOURCE
 extern char *strerror_r(int errnum, char *buf, size_t n);
