diff -Nuar hplip-3.19.3.orig/protocol/discovery/mdns.c hplip-3.19.3/protocol/discovery/mdns.c
--- hplip-3.19.3.orig/protocol/discovery/mdns.c	2019-04-14 16:45:23.721741088 +0200
+++ hplip-3.19.3/protocol/discovery/mdns.c	2019-04-14 16:47:55.066732717 +0200
@@ -39,6 +39,10 @@
 #include "mdns.h"
 #include <unistd.h>
 #include <ctype.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <malloc.h>
+#include <sys/select.h>
 
 /* Convert "www.google.com" to "3www6google3com". */
 static int mdns_convert_name_to_dns(const char *name, int name_size, char *dns_name)
