diff -Nuar elfix-0.9.0.orig/scripts/paxmodule.c elfix-0.9.0/scripts/paxmodule.c
--- elfix-0.9.0.orig/scripts/paxmodule.c	2014-10-13 23:26:32.000000000 +0200
+++ elfix-0.9.0/scripts/paxmodule.c	2014-12-31 19:07:17.470349368 +0100
@@ -101,7 +101,11 @@
 #endif
 
 	if (m == NULL)
+#if PY_MAJOR_VERSION >= 3
+		return NULL;
+#else
 		return;
+#endif
 
 	PaxError = PyErr_NewException("pax.PaxError", NULL, NULL);
 	Py_INCREF(PaxError);
