diff -Nuar qtbase-opensource-src-5.9.2.orig/src/tools/moc/main.cpp qtbase-opensource-src-5.9.2/src/tools/moc/main.cpp
--- qtbase-opensource-src-5.9.2.orig/src/tools/moc/main.cpp	2017-10-02 10:43:38.000000000 +0200
+++ qtbase-opensource-src-5.9.2/src/tools/moc/main.cpp	2017-10-22 12:04:25.053991755 +0200
@@ -267,6 +267,7 @@
     QCommandLineOption prependIncludeOption(QStringLiteral("b"));
     prependIncludeOption.setDescription(QStringLiteral("Prepend #include <file> (preserve default include)."));
     prependIncludeOption.setValueName(QStringLiteral("file"));
+    prependIncludeOption.setFlags(QCommandLineOption::ShortOptionStyle);
     parser.addOption(prependIncludeOption);
 
     QCommandLineOption includeOption(QStringLiteral("include"));
